diff -Nru perl-5.16.0.orig/ext/DynaLoader/Makefile.PL perl-5.16.0/ext/DynaLoader/Makefile.PL
--- perl-5.16.0.orig/ext/DynaLoader/Makefile.PL	Sun May 20 06:00:24 2012
+++ ext/DynaLoader/Makefile.PL	Mon May 28 16:32:45 2012
@@ -64,8 +64,8 @@
 
     return "
 $object : \$(FIRST_MAKEFILE) \$(OBJECT)
-	\$(RM_RF) $object
-	\$(CP) \$(OBJECT) $object
+	#\$(RM_RF) $object
+	#\$(CP) \$(OBJECT) $object
 
 static :: $object
 ";
