diff -Nru ../perl-5.16.0.orig/utils/Makefile.SH ./utils/Makefile.SH
--- ../perl-5.16.0.orig/utils/Makefile.SH	Tue Apr 24 17:18:36 2012
+++ ./utils/Makefile.SH	Wed May 30 20:55:13 2012
@@ -48,9 +48,9 @@
 # Files to be built with variable substitution after miniperl is
 # available.  Dependencies handled manually below (for now).
 
-pl = c2ph.PL config_data.PL corelist.PL cpan.PL h2ph.PL h2xs.PL instmodsh.PL json_pp.PL perlbug.PL perldoc.PL perlivp.PL pl2pm.PL prove.PL ptar.PL ptardiff.PL ptargrep.PL cpanp-run-perl.PL cpanp.PL cpan2dist.PL shasum.PL splain.PL libnetcfg.PL piconv.PL enc2xs.PL xsubpp.PL pod2html.PL zipdetails.PL
-plextract = c2ph config_data corelist cpan h2ph h2xs instmodsh json_pp perlbug perldoc perlivp pl2pm prove ptar ptardiff ptargrep cpanp-run-perl cpanp cpan2dist shasum splain libnetcfg piconv enc2xs xsubpp pod2html zipdetails
-plextractexe = ./c2ph ./config_data ./corelist ./cpan ./h2ph ./h2xs ./json_pp ./instmodsh ./perlbug ./perldoc ./perlivp ./pl2pm ./prove ./ptar ./ptardiff ./ptargrep ./cpanp-run-perl ./cpanp ./cpan2dist ./shasum ./splain ./libnetcfg ./piconv ./enc2xs ./xsubpp ./pod2html ./zipdetails
+pl = c2ph.PL config_data.PL corelist.PL cpan.PL h2ph.PL h2xs.PL instmodsh.PL json_pp.PL perlbug.PL perldoc.PL perlivp.PL pl2pm.PL prove.PL cpanp-run-perl.PL cpanp.PL cpan2dist.PL shasum.PL splain.PL libnetcfg.PL piconv.PL enc2xs.PL xsubpp.PL pod2html.PL zipdetails.PL
+plextract = c2ph config_data corelist cpan h2ph h2xs instmodsh json_pp perlbug perldoc perlivp pl2pm prove cpanp-run-perl cpanp cpan2dist shasum splain libnetcfg piconv enc2xs xsubpp pod2html zipdetails
+plextractexe = ./c2ph ./config_data ./corelist ./cpan ./h2ph ./h2xs ./json_pp ./instmodsh ./perlbug ./perldoc ./perlivp ./pl2pm ./prove ./cpanp-run-perl ./cpanp ./cpan2dist ./shasum ./splain ./libnetcfg ./piconv ./enc2xs ./xsubpp ./pod2html ./zipdetails
 
 all: $(plextract) 
 
@@ -80,12 +80,6 @@
 perlivp:	perlivp.PL ../config.sh
 
 prove:		prove.PL ../config.sh
-
-ptar:		ptar.PL ../config.sh
-
-ptardiff:	ptardiff.PL ../config.sh
-
-ptargrep:	ptargrep.PL ../config.sh
 
 cpanp-run-perl:	cpanp-run-perl.PL ../config.sh
 
diff -Nru ../perl-5.16.0.orig/utils.lst ./utils.lst
--- ../perl-5.16.0.orig/utils.lst	Tue Apr 24 17:18:36 2012
+++ ./utils.lst	Wed May 30 20:55:35 2012
@@ -14,16 +14,13 @@
 utils/instmodsh
 utils/json_pp
 utils/libnetcfg
-utils/perlbug	# link = utils/perlthanks
+utils/perlbug
 utils/perldoc
 utils/perlivp
 utils/piconv
 utils/pl2pm
 utils/pod2html
 utils/prove
-utils/ptar
-utils/ptardiff
-utils/ptargrep
 utils/cpanp-run-perl
 utils/cpanp
 utils/cpan2dist
