diff -Nrux *.orig ../perl-5.16.0.orig/pod/perlop.pod ./pod/perlop.pod
--- ../perl-5.16.0.orig/pod/perlop.pod	Tue Apr 24 19:22:54 2012
+++ ./pod/perlop.pod	Sun Jun  3 12:29:24 2012
@@ -1539,7 +1539,7 @@
 C<"\015">.  If you get in the habit of using C<"\n"> for networking,
 you may be burned some day.
 X<newline> X<line terminator> X<eol> X<end of line>
-X<\n> X<\r> X<\r\n>
+X<\r>
 
 For constructs that do interpolate, variables beginning with "C<$>"
 or "C<@>" are interpolated.  Subscripted variables such as C<$a[3]> or
