diff -x CVS -Nru ../perl-5.12.2/cpan/podlators/lib/Pod/Man.pm ./cpan/podlators/lib/Pod/Man.pm
--- ../perl-5.12.2/cpan/podlators/lib/Pod/Man.pm	Sun Sep  5 08:14:33 2010
+++ ./cpan/podlators/lib/Pod/Man.pm	Fri Sep 24 08:06:55 2010
@@ -825,7 +825,9 @@
                 $cut = $i + 1;
                 $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                 last;
-            }
+            } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                $cut = $i + 1;
+	    }
         }
         if ($cut > 0) {
             splice (@dirs, 0, $cut);
