diff -x CVS -Nru ../perl-5.12.2/cpan/Text-Balanced/lib/Text/Balanced.pm ./cpan/Text-Balanced/lib/Text/Balanced.pm
--- ../perl-5.12.2/cpan/Text-Balanced/lib/Text/Balanced.pm	Sun Sep  5 08:14:33 2010
+++ ./cpan/Text-Balanced/lib/Text/Balanced.pm	Fri Sep 24 08:06:55 2010
@@ -1508,7 +1508,7 @@
 If the string is "PARA", C<extract_tagged> returns only the first paragraph
 after the tag (up to the first line that is either empty or contains
 only whitespace characters).
-If the string is "", the the default behaviour (i.e. failure) is reinstated.
+If the string is "", the default behaviour (i.e. failure) is reinstated.
 
 For example, suppose the start tag "/para" introduces a paragraph, which then
 continues until the next "/endpara" tag or until another "/para" tag is
