diff -x CVS -Nru ../perl-5.12.2/dist/Pod-Perldoc/lib/Pod/Perldoc.pm ./dist/Pod-Perldoc/lib/Pod/Perldoc.pm
--- ../perl-5.12.2/dist/Pod-Perldoc/lib/Pod/Perldoc.pm	Sun Sep  5 08:14:33 2010
+++ ./dist/Pod-Perldoc/lib/Pod/Perldoc.pm	Fri Sep 24 08:06:56 2010
@@ -499,7 +499,7 @@
       } else {
         $^W = 0;
         # The average user just has no reason to be seeing
-        #  $^W-suppressable warnings from the the require!
+        #  $^W-suppressible warnings from the require!
       }
 
       eval "require $c";
