diff -x CVS -Nru ../perl-5.12.2/installperl ./installperl
--- ../perl-5.12.2/installperl	Sun Sep  5 08:14:33 2010
+++ ./installperl	Fri Sep 24 08:06:40 2010
@@ -200,7 +200,7 @@
 my $installarchlib = "$opts{destdir}$Config{installarchlib}";
 my $installsitelib = "$opts{destdir}$Config{installsitelib}";
 my $installsitearch = "$opts{destdir}$Config{installsitearch}";
-my $installman1dir = "$opts{destdir}$Config{installman1dir}";
+my $installman1dir = "none";
 my $man1ext = $Config{man1ext};
 my $libperl = $Config{libperl};
 # Shared library and dynamic loading suffixes.
