diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Harness/lib/TAP/Formatter/Console/ParallelSession.pm ./cpan/Test-Harness/lib/TAP/Formatter/Console/ParallelSession.pm
--- ../perl-5.12.2/cpan/Test-Harness/lib/TAP/Formatter/Console/ParallelSession.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Harness/lib/TAP/Formatter/Console/ParallelSession.pm	Fri Sep 24 08:06:54 2010
@@ -55,10 +55,6 @@
 
 =head1 SYNOPSIS
 
-=cut
-
-=head1 METHODS
-
 =head2 Class Methods
 
 =head3 C<header>
