diff -x CVS -Nru ../perl-5.12.2/cpan/Compress-Raw-Bzip2/lib/Compress/Raw/Bzip2.pm ./cpan/Compress-Raw-Bzip2/lib/Compress/Raw/Bzip2.pm
--- ../perl-5.12.2/cpan/Compress-Raw-Bzip2/lib/Compress/Raw/Bzip2.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/Compress-Raw-Bzip2/lib/Compress/Raw/Bzip2.pm	Sat Sep 25 05:24:13 2010
@@ -12,7 +12,7 @@
 use bytes ;
 our ($VERSION, $XS_VERSION, @ISA, @EXPORT, $AUTOLOAD);
 
-$VERSION = '2.024';
+$VERSION = '2.031';
 $XS_VERSION = $VERSION; 
 $VERSION = eval $VERSION;
 
@@ -220,7 +220,7 @@
 
 =head1 Uncompression
 
-=head2 ($z, $status) = new Compress::Raw::Bunzip2 $appendOutput, $consumeInput, $small, $limitOutput;
+=head2 ($z, $status) = new Compress::Raw::Bunzip2 $appendOutput, $consumeInput, $small, $verbosity, $limitOutput;
 
 If successful, it will return the initialised uncompression object, C<$z>
 and a C<$status> of C<BZ_OK> in a list context. In scalar context it
@@ -272,6 +272,12 @@
 enabled.
 
 This option defaults to false.
+
+=item B<$verbosity>
+
+This parameter is ignored.
+
+Defaults to 0.
 
 =back
 
