diff -x CVS -Nru ../perl-5.12.2/cpan/Digest/t/digest.t ./cpan/Digest/t/digest.t
--- ../perl-5.12.2/cpan/Digest/t/digest.t	Sun Sep  5 08:14:32 2010
+++ ./cpan/Digest/t/digest.t	Tue Oct 11 11:13:35 2011
@@ -1,36 +1,22 @@
-print "1..3\n";
+#!/usr/bin/env perl
 
-use Digest;
+use strict;
+use Test::More tests => 4;
 
-{
-    package Digest::Dummy;
-    use vars qw($VERSION @ISA);
-    $VERSION = 1;
+# To find Digest::Dummy
+use lib 't/lib';
 
-    require Digest::base;
-    @ISA = qw(Digest::base);
+use Digest;
 
-    sub new {
-	my $class = shift;
-	my $d = shift || "ooo";
-	bless { d => $d }, $class;
-    }
-    sub add {}
-    sub digest { shift->{d} }
-}
-
+$@ = "rt#50663";
 my $d;
 $d = Digest->new("Dummy");
-print "not " unless $d->digest eq "ooo";
-print "ok 1\n";
+is $@, "rt#50663";
+is $d->digest, "ooo";
 
 $d = Digest->Dummy;
-print "not " unless $d->digest eq "ooo";
-print "ok 2\n";
+is $d->digest, "ooo";
 
 $Digest::MMAP{"Dummy-24"} = [["NotThere"], "NotThereEither", ["Digest::Dummy", 24]];
 $d = Digest->new("Dummy-24");
-print "not " unless $d->digest eq "24";
-print "ok 3\n";
-
-
+is $d->digest, "24";
