diff -x CVS -Nru ../perl-5.12.2/cpan/List-Util/lib/List/Util.pm ./cpan/List-Util/lib/List/Util.pm
--- ../perl-5.12.2/cpan/List-Util/lib/List/Util.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/List-Util/lib/List/Util.pm	Mon Apr 18 01:27:57 2011
@@ -14,7 +14,7 @@
 
 @ISA        = qw(Exporter);
 @EXPORT_OK  = qw(first min max minstr maxstr reduce sum shuffle);
-$VERSION    = "1.22";
+$VERSION    = "1.23";
 $XS_VERSION = $VERSION;
 $VERSION    = eval $VERSION;
 
