diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/Tester/tbt_06errormess.t ./cpan/Test-Simple/t/Tester/tbt_06errormess.t
--- ../perl-5.12.2/cpan/Test-Simple/t/Tester/tbt_06errormess.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/Tester/tbt_06errormess.t	Fri Jan  7 05:19:32 2011
@@ -50,7 +50,7 @@
     $out->reset();
     $err->reset();
 
-    # remeber that we're testing
+    # remember that we're testing
     $testing_num = $t->current_test;
     $t->current_test(0);
 }
