diff -x CVS -Nru ../perl-5.12.2/cpan/CGI/t/http.t ./cpan/CGI/t/http.t
--- ../perl-5.12.2/cpan/CGI/t/http.t	Sun Sep  5 08:14:32 2010
+++ ./cpan/CGI/t/http.t	Thu Jan  6 14:23:16 2011
@@ -34,8 +34,11 @@
     # https()
     # The same as http(), but operates on the HTTPS environment variables present when the SSL protocol is in
     # effect.  Can be used to determine whether SSL is turned on.
+    my @expect = grep /^HTTPS/, keys %ENV;
+    push @expect, 'HTTPS'         if not exists $ENV{HTTPS};
+    push @expect, 'HTTPS_KEYSIZE' if not exists $ENV{HTTPS_KEYSIZE};
     local $ENV{'HTTPS'} = 'ON';
     local $ENV{'HTTPS_KEYSIZE'} = 512;
     is $cgi->https(), 'ON', 'scalar context to check SSL is on';
-    ok eq_set( [$cgi->https()], [qw(HTTPS HTTPS_KEYSIZE)]), 'list context returns https keys';
+    ok eq_set( [$cgi->https()], \@expect), 'list context returns https keys';
 }
