diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Dist/Autobundle.pm ./cpan/CPANPLUS/lib/CPANPLUS/Dist/Autobundle.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Dist/Autobundle.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Dist/Autobundle.pm	Fri Sep 24 08:06:45 2010
@@ -10,7 +10,7 @@
 
 =head1 NAME
 
-CPANPLUS::Dist::Autobundle
+CPANPLUS::Dist::Autobundle - class for installing installation snapshots
 
 =head1 SYNOPSIS
 
