diff -x CVS -Nru ../perl-5.12.2/cpan/Encode/Encode.pm ./cpan/Encode/Encode.pm
--- ../perl-5.12.2/cpan/Encode/Encode.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/Encode/Encode.pm	Fri Sep 24 08:06:46 2010
@@ -913,7 +913,7 @@
 
 Do you copy?  As of Perl 5.8.7, B<UTF-8> means strict, official UTF-8
 while B<utf8> means liberal, lax, version thereof.  And Encode version
-2.10 or later thus groks the difference between C<UTF-8> and C"utf8".
+2.10 or later thus groks the difference between C<UTF-8> and C<utf8>.
 
   encode("utf8",  "\x{FFFF_FFFF}", 1); # okay
   encode("UTF-8", "\x{FFFF_FFFF}", 1); # croaks
