diff -x CVS -Nru ../perl-5.12.2/cpan/List-Util/lib/List/Util/XS.pm ./cpan/List-Util/lib/List/Util/XS.pm
--- ../perl-5.12.2/cpan/List-Util/lib/List/Util/XS.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/List-Util/lib/List/Util/XS.pm	Mon Apr 18 01:27:57 2011
@@ -3,7 +3,7 @@
 use vars qw($VERSION);
 use List::Util;
 
-$VERSION = "1.22";           # FIXUP
+$VERSION = "1.23";           # FIXUP
 $VERSION = eval $VERSION;    # FIXUP
 
 sub _VERSION { # FIXUP
