diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Module/Checksums.pm ./cpan/CPANPLUS/lib/CPANPLUS/Module/Checksums.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Module/Checksums.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Module/Checksums.pm	Fri Sep 24 08:06:45 2010
@@ -19,7 +19,7 @@
 
 =head1 NAME
 
-CPANPLUS::Module::Checksums
+CPANPLUS::Module::Checksums - distribution checksum validation
 
 =head1 SYNOPSIS
 
