diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/lib/Test/Builder/Tester.pm ./cpan/Test-Simple/lib/Test/Builder/Tester.pm
--- ../perl-5.12.2/cpan/Test-Simple/lib/Test/Builder/Tester.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/lib/Test/Builder/Tester.pm	Mon Apr 18 10:48:24 2011
@@ -1,7 +1,7 @@
 package Test::Builder::Tester;
 
 use strict;
-our $VERSION = "1.18";
+our $VERSION = "1.22";
 
 use Test::Builder;
 use Symbol;
@@ -124,13 +124,13 @@
     # switch out to our own handles
     $t->output($output_handle);
     $t->failure_output($error_handle);
-    $t->todo_output($error_handle);
+    $t->todo_output($output_handle);
 
     # clear the expected list
     $out->reset();
     $err->reset();
 
-    # remeber that we're testing
+    # remember that we're testing
     $testing     = 1;
     $testing_num = $t->current_test;
     $t->current_test(0);
@@ -165,8 +165,8 @@
    test_out("ok 2");
 
 Once C<test_out> or C<test_err> (or C<test_fail> or C<test_diag>) have
-been called once all further output from B<Test::Builder> will be
-captured by B<Test::Builder::Tester>.  This means that your will not
+been called, all further output from B<Test::Builder> will be
+captured by B<Test::Builder::Tester>.  This means that you will not
 be able perform further tests to the normal output in the normal way
 until you call C<test_test> (well, unless you manually meddle with the
 output filehandles)
@@ -191,7 +191,7 @@
 
 Because the standard failure message that B<Test::Builder> produces
 whenever a test fails will be a common occurrence in your test error
-output, and because has changed between Test::Builder versions, rather
+output, and because it has changed between Test::Builder versions, rather
 than forcing you to call C<test_err> with the string all the time like
 so
 
@@ -229,7 +229,7 @@
 
 As most of the remaining expected output to the error stream will be
 created by Test::Builder's C<diag> function, B<Test::Builder::Tester>
-provides a convience function C<test_diag> that you can use instead of
+provides a convenience function C<test_diag> that you can use instead of
 C<test_err>.
 
 The C<test_diag> function prepends comment hashes and spacing to the
@@ -293,7 +293,7 @@
 
 =back
 
-As a convience, if only one argument is passed then this argument
+As a convenience, if only one argument is passed then this argument
 is assumed to be the name of the test (as in the above examples.)
 
 Once C<test_test> has been run test output will be redirected back to
@@ -304,7 +304,7 @@
 =cut
 
 sub test_test {
-    # decode the arguements as described in the pod
+    # decode the arguments as described in the pod
     my $mess;
     my %args;
     if( @_ == 1 ) {
@@ -370,7 +370,7 @@
 
 =back
 
-In addition to the six exported functions there there exists one
+In addition to the six exported functions there exists one
 function that can only be accessed with a fully qualified function
 call.
 
@@ -389,7 +389,7 @@
 
 To assist you C<test_test> can colour the background of the debug
 information to disambiguate the different types of output. The debug
-output will have it's background coloured green and red.  The green
+output will have its background coloured green and red.  The green
 part represents the text which is the same between the executed and
 actual output, the red shows which part differs.
 
