diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Internals/Source.pm ./cpan/CPANPLUS/lib/CPANPLUS/Internals/Source.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Internals/Source.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Internals/Source.pm	Fri Sep 24 08:06:45 2010
@@ -74,7 +74,7 @@
 
 =head1 NAME
 
-CPANPLUS::Internals::Source
+CPANPLUS::Internals::Source - updating and parsing of source files for CPANPLUS
 
 =head1 SYNOPSIS
 
