diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/Builder/no_plan_at_all.t ./cpan/Test-Simple/t/Builder/no_plan_at_all.t
--- ../perl-5.12.2/cpan/Test-Simple/t/Builder/no_plan_at_all.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/Builder/no_plan_at_all.t	Fri Jan  7 05:19:32 2011
@@ -1,6 +1,6 @@
 #!/usr/bin/perl -w
 
-# Test what happens when no plan is delcared and done_testing() is not seen
+# Test what happens when no plan is declared and done_testing() is not seen
 
 use strict;
 BEGIN {
