diff -x CVS -Nru ../perl-5.12.2/cpan/Pod-Parser/lib/Pod/Usage.pm ./cpan/Pod-Parser/lib/Pod/Usage.pm
--- ../perl-5.12.2/cpan/Pod-Parser/lib/Pod/Usage.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/Pod-Parser/lib/Pod/Usage.pm	Fri Sep 24 08:06:54 2010
@@ -16,7 +16,7 @@
 
 =head1 NAME
 
-Pod::Usage, pod2usage() - print a usage message from embedded pod documentation
+Pod::Usage, pod2usage - print a usage message from embedded pod documentation
 
 =head1 SYNOPSIS
 
