diff -x CVS -Nru ../perl-5.12.2/cpan/Compress-Raw-Bzip2/Changes ./cpan/Compress-Raw-Bzip2/Changes
--- ../perl-5.12.2/cpan/Compress-Raw-Bzip2/Changes	Mon Sep  6 16:30:32 2010
+++ ./cpan/Compress-Raw-Bzip2/Changes	Sat Sep 25 05:24:13 2010
@@ -1,5 +1,28 @@
 CHANGES
 -------
+
+  2.031 21 September 2010
+
+      * Updated to use bzip2 1.0.6
+
+  2.030 22 July 2010
+
+      * No Changes
+
+  2.027 24 April 2010
+
+      * No Changes
+
+  2.026 7 April 2010
+
+      * No Changes
+
+  2.025 27 March 2010
+
+      * Documented the unused "verbosity" option in
+        Compress::Raw::Bunzip2::new
+        [RT# 54425]
+
   2.023 9 November 2009
 
       * Removed redundant bzip2 source files from the bzip2-src directory.
