diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/Builder/reset.t ./cpan/Test-Simple/t/Builder/reset.t
--- ../perl-5.12.2/cpan/Test-Simple/t/Builder/reset.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/Builder/reset.t	Fri Jan  7 05:19:32 2011
@@ -44,11 +44,13 @@
 $tb->no_header(1);
 $tb->no_ending(1);
 
+$tb->done_testing;  # make sure done_testing gets reset
 
 # Now reset it.
 $tb->reset;
 
 
+# Test the state of the reset builder
 $Test->ok( !defined $tb->exported_to, 'exported_to' );
 $Test->is_eq( $tb->expected_tests, 0, 'expected_tests' );
 $Test->is_eq( $tb->level,          1, 'level' );
@@ -65,9 +67,12 @@
 $Test->is_eq( fileno $tb->todo_output,
               fileno $Original_Output{todo_output},    'todo_output' );
 
-$tb->current_test(12);
+# The reset Test::Builder will take over from here.
+$Test->no_ending(1);
+
+
+$tb->current_test($Test->current_test);
 $tb->level(0);
 $tb->ok(1, 'final test to make sure output was reset');
 
-$Test->current_test(13);
-$Test->done_testing(13);
+$tb->done_testing;
