diff -x CVS -Nru ../perl-5.12.2/cpan/CGI/lib/CGI/Push.pm ./cpan/CGI/lib/CGI/Push.pm
--- ../perl-5.12.2/cpan/CGI/lib/CGI/Push.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/CGI/lib/CGI/Push.pm	Thu Jan  6 14:23:16 2011
@@ -16,7 +16,7 @@
 # The most recent version and complete docs are available at:
 #   http://stein.cshl.org/WWW/software/CGI/
 
-$CGI::Push::VERSION='1.04';
+$CGI::Push::VERSION='1.05';
 use CGI;
 use CGI::Util 'rearrange';
 @ISA = ('CGI');
@@ -214,7 +214,7 @@
 
 This optional parameter indicates the content type of each page.  It
 defaults to "text/html".  Normally the module assumes that each page
-is of a homogenous MIME type.  However if you provide either of the
+is of a homogeneous MIME type.  However if you provide either of the
 magic values "heterogeneous" or "dynamic" (the latter provided for the
 convenience of those who hate long parameter names), you can specify
 the MIME type -- and other header fields -- on a per-page basis.  See 
