diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/lib/Test/Simple.pm ./cpan/Test-Simple/lib/Test/Simple.pm
--- ../perl-5.12.2/cpan/Test-Simple/lib/Test/Simple.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/lib/Test/Simple.pm	Mon Apr 18 10:48:24 2011
@@ -4,7 +4,7 @@
 
 use strict;
 
-our $VERSION = '0.94';
+our $VERSION = '0.98';
 $VERSION = eval $VERSION;    ## no critic (BuiltinFunctions::ProhibitStringyEval)
 
 use Test::Builder::Module;
@@ -121,7 +121,7 @@
                              Rating   => 'R',
                              NumExplodingSheep => 1
                            });
-    ok( defined($btaste) && ref $btaste eq 'Film,     'new() works' );
+    ok( defined($btaste) && ref $btaste eq 'Film',     'new() works' );
 
     ok( $btaste->Title      eq 'Bad Taste',     'Title() get'    );
     ok( $btaste->Director   eq 'Peter Jackson', 'Director() get' );
