diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm ./cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm	Fri Sep 24 08:06:45 2010
@@ -19,7 +19,7 @@
 
 =head1 NAME
 
-CPANPLUS::Internals::Report
+CPANPLUS::Internals::Report -  send test reports to http://testers.cpan.org
 
 =head1 SYNOPSIS
 
