diff -x CVS -Nru ../perl-5.12.2/cpan/Pod-Parser/lib/Pod/Select.pm ./cpan/Pod-Parser/lib/Pod/Select.pm
--- ../perl-5.12.2/cpan/Pod-Parser/lib/Pod/Select.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/Pod-Parser/lib/Pod/Select.pm	Fri Sep 24 08:06:54 2010
@@ -18,7 +18,7 @@
 
 =head1 NAME
 
-Pod::Select, podselect() - extract selected sections of POD from input
+Pod::Select, podselect - extract selected sections of POD from input
 
 =head1 SYNOPSIS
 
