diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/Builder/no_ending.t ./cpan/Test-Simple/t/Builder/no_ending.t
--- ../perl-5.12.2/cpan/Test-Simple/t/Builder/no_ending.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/Builder/no_ending.t	Fri Jan  7 05:19:32 2011
@@ -15,7 +15,7 @@
 use Test::More tests => 3;
 
 # Normally, Test::More would yell that we ran too few tests, but we
-# supressed the ending diagnostics.
+# suppressed the ending diagnostics.
 pass;
 print "ok 2\n";
 print "ok 3\n";
