diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Config.pm ./cpan/CPANPLUS/lib/CPANPLUS/Config.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Config.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Config.pm	Fri Sep 24 08:06:44 2010
@@ -24,7 +24,7 @@
 
 =head1 NAME
 
-CPANPLUS::Config
+CPANPLUS::Config - defaults and heuristics for CPANPLUS configuration information
 
 =head1 SYNOPSIS
 
