diff -x CVS -Nru ../perl-5.12.2/cpan/List-Util/lib/Scalar/Util.pm ./cpan/List-Util/lib/Scalar/Util.pm
--- ../perl-5.12.2/cpan/List-Util/lib/Scalar/Util.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/List-Util/lib/Scalar/Util.pm	Mon Apr 18 01:27:57 2011
@@ -13,7 +13,7 @@
 
 @ISA       = qw(Exporter);
 @EXPORT_OK = qw(blessed dualvar reftype weaken isweak tainted readonly openhandle refaddr isvstring looks_like_number set_prototype);
-$VERSION    = "1.22";
+$VERSION    = "1.23";
 $VERSION   = eval $VERSION;
 
 unless (defined &dualvar) {
