diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Dist/MM.pm ./cpan/CPANPLUS/lib/CPANPLUS/Dist/MM.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Dist/MM.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Dist/MM.pm	Fri Sep 24 08:06:45 2010
@@ -23,7 +23,7 @@
 
 =head1 NAME
 
-CPANPLUS::Dist::MM
+CPANPLUS::Dist::MM - distribution class for MakeMaker related modules
 
 =head1 SYNOPSIS
 
