diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Module/Author/Fake.pm ./cpan/CPANPLUS/lib/CPANPLUS/Module/Author/Fake.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Module/Author/Fake.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Module/Author/Fake.pm	Fri Sep 24 08:06:45 2010
@@ -17,7 +17,7 @@
 
 =head1 NAME
 
-CPANPLUS::Module::Author::Fake
+CPANPLUS::Module::Author::Fake - class for creating fake author objects
 
 =head1 SYNOPSIS
 
