diff -x CVS -Nru ../perl-5.12.2/cpan/Compress-Raw-Bzip2/bzip2-src/LICENSE ./cpan/Compress-Raw-Bzip2/bzip2-src/LICENSE
--- ../perl-5.12.2/cpan/Compress-Raw-Bzip2/bzip2-src/LICENSE	Sun Sep  5 08:14:32 2010
+++ ./cpan/Compress-Raw-Bzip2/bzip2-src/LICENSE	Sat Sep 25 05:24:13 2010
@@ -2,7 +2,7 @@
 --------------------------------------------------------------------------
 
 This program, "bzip2", the associated library "libbzip2", and all
-documentation, are copyright (C) 1996-2007 Julian R Seward.  All
+documentation, are copyright (C) 1996-2010 Julian R Seward.  All
 rights reserved.
 
 Redistribution and use in source and binary forms, with or without
@@ -37,6 +37,6 @@
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 Julian Seward, jseward@bzip.org
-bzip2/libbzip2 version 1.0.5 of 10 December 2007
+bzip2/libbzip2 version 1.0.6 of 6 September 2010
 
 --------------------------------------------------------------------------
