diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/Tester/tbt_01basic.t ./cpan/Test-Simple/t/Tester/tbt_01basic.t
--- ../perl-5.12.2/cpan/Test-Simple/t/Tester/tbt_01basic.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/Tester/tbt_01basic.t	Fri Jan  7 05:19:32 2011
@@ -46,7 +46,7 @@
 
 
 test_out("not ok 1 - name # TODO Something");
-test_err("#     Failed (TODO) test ($0 at line 52)");
+test_out("#     Failed (TODO) test ($0 at line 52)");
 TODO: { 
     local $TODO = "Something";
     fail("name");
