diff -x CVS -Nru ../perl-5.12.2/cpan/Module-Build/lib/Module/Build/ModuleInfo.pm ./cpan/Module-Build/lib/Module/Build/ModuleInfo.pm
--- ../perl-5.12.2/cpan/Module-Build/lib/Module/Build/ModuleInfo.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/Module-Build/lib/Module/Build/ModuleInfo.pm	Fri Sep 24 08:06:53 2010
@@ -395,7 +395,7 @@
 
 =head1 NAME
 
-ModuleInfo - Gather package and POD information from a perl module file
+Module::Build::ModuleInfo - Gather package and POD information from a perl module file
 
 
 =head1 DESCRIPTION
