diff -x CVS -Nru ../perl-5.12.2/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm ./cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
--- ../perl-5.12.2/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm	Mon Sep  6 16:30:32 2010
+++ ./cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm	Fri Sep 24 08:06:50 2010
@@ -289,7 +289,7 @@
     return '' unless $self->needs_linking();
     return $self->{CONST_CCCMD} =
 	q{CCCMD = $(CC) -c $(PASTHRU_INC) $(INC) \\
-	$(CCFLAGS) $(OPTIMIZE) \\
+	$(CCFLAGS) $(OPTIMIZE) $(COPTS) \\
 	$(PERLTYPE) $(MPOLLUTE) $(DEFINE_VERSION) \\
 	$(XS_DEFINE_VERSION)};
 }
