diff -x CVS -Nru ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Internals/Search.pm ./cpan/CPANPLUS/lib/CPANPLUS/Internals/Search.pm
--- ../perl-5.12.2/cpan/CPANPLUS/lib/CPANPLUS/Internals/Search.pm	Sun Sep  5 08:14:32 2010
+++ ./cpan/CPANPLUS/lib/CPANPLUS/Internals/Search.pm	Fri Sep 24 08:06:45 2010
@@ -19,7 +19,7 @@
 
 =head1 NAME
 
-CPANPLUS::Internals::Search
+CPANPLUS::Internals::Search - find CPAN modules
 
 =head1 SYNOPSIS
 
