diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/use_ok.t ./cpan/Test-Simple/t/use_ok.t
--- ../perl-5.12.2/cpan/Test-Simple/t/use_ok.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/use_ok.t	Fri Jan  7 05:19:31 2011
@@ -22,7 +22,7 @@
 {
     package Foo::two;
     ::use_ok("Symbol", qw(qualify));
-    ::ok( !defined &gensym,       '  one arg, defaults overriden' );
+    ::ok( !defined &gensym,       '  one arg, defaults overridden' );
     ::ok( defined &qualify,       '  right function exported' );
 }
 
