diff -x CVS -Nru ../perl-5.12.2/cpan/Test-Simple/t/dependents.t ./cpan/Test-Simple/t/dependents.t
--- ../perl-5.12.2/cpan/Test-Simple/t/dependents.t	Mon Sep  6 16:30:32 2010
+++ ./cpan/Test-Simple/t/dependents.t	Fri Jan  7 05:19:31 2011
@@ -11,6 +11,7 @@
     plan skip_all => "Dependents only tested when releasing" unless $ENV{PERL_RELEASING};
 }
 
+require File::Spec;
 use CPAN;
 
 CPAN::HandleConfig->load;
@@ -24,6 +25,7 @@
     Test::Class
     Test::Deep
     Test::Differences
+    Test::NoWarnings
 );
 
 # Modules which are known to be broken
@@ -34,6 +36,7 @@
 TODO: for my $name (@ARGV ? @ARGV : @Modules) {
     local $TODO = "$name known to be broken" if $Broken{$name};
 
+    local $ENV{PERL5LIB} = File::Spec->rel2abs("blib/lib");
     my $module = CPAN::Shell->expand("Module", $name);
     $module->test;
     ok( !$module->distribution->{make_test}->failed, $name );
