diff -x CVS -Nru ../perl-5.12.2/Cross/README ./Cross/README
--- ../perl-5.12.2/Cross/README	Sun Sep  5 08:14:32 2010
+++ ./Cross/README	Fri Sep 24 08:06:43 2010
@@ -47,7 +47,7 @@
 ------------------------------
 
 It is possible to extend the cross-compilation to other targets.
-We have sucessfully compiled for the target solaris2.8/x86
+We have successfully compiled for the target solaris2.8/x86
 on linux/x86 build system.
 
 To attempt a cross-compile for another target using the methods
