diff -x CVS -Nru ../perl-5.12.2/ext/PerlIO-encoding/encoding.xs ./ext/PerlIO-encoding/encoding.xs
--- ../perl-5.12.2/ext/PerlIO-encoding/encoding.xs	Sun Sep  5 08:14:33 2010
+++ ./ext/PerlIO-encoding/encoding.xs	Fri Sep 24 08:06:57 2010
@@ -540,8 +540,8 @@
 PerlIOEncode_tell(pTHX_ PerlIO * f)
 {
     PerlIOBuf *b = PerlIOSelf(f, PerlIOBuf);
-    /* Unfortunately the only way to get a postion is to (re-)translate,
-       the UTF8 we have in bufefr and then ask layer below
+    /* Unfortunately the only way to get a position is to (re-)translate,
+       the UTF8 we have in buffer and then ask layer below
      */
     PerlIO_flush(f);
     if (b->buf && b->ptr > b->buf) {
