diff -Nru perl-5.16.0.orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Source/Memory.pm perl-5.16.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Source/Memory.pm
--- perl-5.16.0.orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Source/Memory.pm	Tue Apr 24 17:18:30 2012
+++ cpan/CPANPLUS/lib/CPANPLUS/Internals/Source/Memory.pm	Mon May 28 16:32:40 2012
@@ -191,6 +191,8 @@
 
 =pod
 
+=head1 SYNOPSIS
+
 =head2 $cb->__memory_retrieve_source(name => $name, [path => $path, uptodate => BOOL, verbose => BOOL])
 
 This method retrieves a I<storable>d tree identified by C<$name>.
