diff -Nru perl-5.16.0.orig/cpan/NEXT/lib/NEXT.pm perl-5.16.0/cpan/NEXT/lib/NEXT.pm
--- perl-5.16.0.orig/cpan/NEXT/lib/NEXT.pm	Tue Feb 14 21:44:41 2012
+++ cpan/NEXT/lib/NEXT.pm	Mon May 28 16:32:44 2012
@@ -190,7 +190,7 @@
 
 =head1 NAME
 
-NEXT.pm - Provide a pseudo-class NEXT (et al) that allows method redispatch
+NEXT - Provide a pseudo-class NEXT (et al) that allows method redispatch
 
 
 =head1 SYNOPSIS
