diff -Nru perl-5.16.0.orig/dist/ExtUtils-ParseXS/lib/perlxstut.pod perl-5.16.0/dist/ExtUtils-ParseXS/lib/perlxstut.pod
--- perl-5.16.0.orig/dist/ExtUtils-ParseXS/lib/perlxstut.pod	Tue Apr 24 17:18:33 2012
+++ dist/ExtUtils-ParseXS/lib/perlxstut.pod	Mon May 28 16:32:55 2012
@@ -244,7 +244,7 @@
 Makefile.PL file, and running make.
 
 In order to test that our extension works, we now need to look at the
-file Mytest.t.  This file is set up to imitate the same kind of testing
+file t/Mytest.t.  This file is set up to imitate the same kind of testing
 structure that Perl itself has.  Within the test script, you perform a
 number of tests to confirm the behavior of the extension, printing "ok"
 when the test is correct, "not ok" when it is not.
