diff -Nru perl-5.16.0.orig/hints/openbsd.sh perl-5.16.0/hints/openbsd.sh
--- perl-5.16.0.orig/hints/openbsd.sh	Tue Feb 14 21:44:48 2012
+++ hints/openbsd.sh	Mon May 28 16:32:50 2012
@@ -15,11 +15,6 @@
     ;;
 esac
 
-# malloc wrap works
-case "$usemallocwrap" in
-'') usemallocwrap='define' ;;
-esac
-
 # Currently, vfork(2) is not a real win over fork(2).
 usevfork="$undef"
 
@@ -44,7 +39,7 @@
 #
 ARCH=`arch | sed 's/^OpenBSD.//'`
 case "${ARCH}-${osvers}" in
-alpha-2.[0-8]|mips-2.[0-8]|powerpc-2.[0-7]|m88k-*|hppa-*|vax-*)
+alpha-2.[0-8]|mips-2.[0-8]|powerpc-2.[0-7]|m88k-*|hppa-3.[0-5]|vax-*)
 	test -z "$usedl" && usedl=$undef
 	;;
 *)
@@ -84,6 +79,14 @@
 	;;
 esac
 
+# malloc wrap causes problems on m68k
+if [ X"$usemallocwrap" = X"" ]; then
+	case "${ARCH}" in
+	m68k) usemallocwrap="$undef" ;;
+	*)    usemallocwrap="define" ;;
+	esac
+fi
+
 # OpenBSD doesn't need libcrypt but many folks keep a stub lib
 # around for old NetBSD binaries.
 libswanted=`echo $libswanted | sed 's/ crypt / /'`
@@ -93,12 +96,12 @@
 
 # cc is gcc so we can do better than -O
 # Allow a command-line override, such as -Doptimize=-g
-case ${ARCH} in
-m88k)
-   optimize='-O0'
+case "${ARCH}-${osvers}" in
+hppa-3.3|m88k-2.*|m88k-3.[0-3])
+   test "$optimize" || optimize='-O0'
    ;;
-hppa)
-   optimize='-O0'
+m88k-3.4)
+   test "$optimize" || optimize='-O1'
    ;;
 *)
    test "$optimize" || optimize='-O2'
@@ -122,8 +125,8 @@
 	;;
 	esac
 	case "$osvers" in
-	[012].*|3.[0-6])
-        	# Broken at least up to OpenBSD 3.6, we'll see about 3.7
+	[012].*|3.[0-5])
+        	# Broken up to OpenBSD 3.6, fixed in OpenBSD 3.7
 		d_getservbyname_r=$undef ;;
 	esac
 esac
@@ -206,9 +209,9 @@
 	locincpth='/usr/local/include'
 	loclibpth='/usr/local/lib'
 	# Link perl with shared libperl
-	if [ "$usedl" = "$define" -a -r shlib_version ]; then
+	if [ "$usedl" = "$define" -a -r $src/shlib_version ]; then
 		useshrplib=true
-		libperl=`. ./shlib_version; echo libperl.so.${major}.${minor}`
+		libperl=`. $src/shlib_version; echo libperl.so.${major}.${minor}`
 	fi
 	;;
 esac
