diff -Nru perl-5.16.0.orig/cpan/Pod-Parser/lib/Pod/PlainText.pm perl-5.16.0/cpan/Pod-Parser/lib/Pod/PlainText.pm
--- perl-5.16.0.orig/cpan/Pod-Parser/lib/Pod/PlainText.pm	Tue Apr 24 17:18:31 2012
+++ cpan/Pod-Parser/lib/Pod/PlainText.pm	Mon May 28 16:32:44 2012
@@ -592,6 +592,11 @@
     }
 }
 
+sub begin_pod {
+    my $self = shift;
+    $$self{EXCLUDE} = 0;
+    $$self{VERBATIM} = 0;
+}
 
 ############################################################################
 # Module return value and documentation
