diff -Nru perl-5.16.0.orig/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm perl-5.16.0/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
--- perl-5.16.0.orig/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm	Tue Apr 24 17:18:31 2012
+++ cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm	Mon May 28 16:32:44 2012
@@ -290,7 +290,7 @@
     return '' unless $self->needs_linking();
     return $self->{CONST_CCCMD} =
 	q{CCCMD = $(CC) -c $(PASTHRU_INC) $(INC) \\
-	$(CCFLAGS) $(OPTIMIZE) \\
+	$(CCFLAGS) $(OPTIMIZE) $(COPTS) \\
 	$(PERLTYPE) $(MPOLLUTE) $(DEFINE_VERSION) \\
 	$(XS_DEFINE_VERSION)};
 }
