diff -Nru perl-5.16.0.orig/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm perl-5.16.0/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm
--- perl-5.16.0.orig/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm	Tue Apr 24 17:18:31 2012
+++ cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm	Mon May 28 16:32:44 2012
@@ -774,8 +774,9 @@
     my @man_cmds;
     foreach my $section (qw(1 3)) {
         my $pods = $self->{"MAN${section}PODS"};
+	my $s = $section eq '3' ? '3p' : $section;
         push @man_cmds, $self->split_command(<<CMD, %$pods);
-	\$(NOECHO) \$(POD2MAN) --section=$section --perm_rw=\$(PERM_RW)
+	\$(NOECHO) \$(POD2MAN) --section=$s --perm_rw=\$(PERM_RW)
 CMD
     }
 
