diff -Nru perl-5.16.0.orig/MANIFEST perl-5.16.0/MANIFEST
--- perl-5.16.0.orig/MANIFEST	Mon May 14 12:49:44 2012
+++ MANIFEST	Mon May 28 16:32:09 2012
@@ -10,6 +10,7 @@
 Changes			Describe how to peruse changes between releases
 charclass_invlists.h	Compiled-in inversion lists
 config_h.SH		Produces config.h
+config.over		Site-specific overrides for Configure defaults
 configpm		Produces lib/Config.pm
 Configure		Portability tool
 configure.com		Configure-equivalent for VMS
@@ -4060,6 +4061,7 @@
 form.h				Public declarations for formats
 generate_uudmap.c		Generate uudmap.h, the uuencode decoding map
 genpacksizetables.pl		Generate the size tables for pack/unpack
+git_version.h			Pre-generated git_version.h for OpenBSD
 globals.c			File to declare global symbols (for shared library)
 globvar.sym			Global variables that need hiding when embedded
 gv.c				Glob value code
@@ -4205,6 +4207,7 @@
 lib/Class/Struct.t		See if Class::Struct works
 lib/Config/Extensions.pm	Convenient hash lookup for built extensions
 lib/Config/Extensions.t		See if Config::Extensions works
+lib/Config_git.pl		Pre-generated git_version.h for OpenBSD
 lib/Config.t			See if Config works
 lib/CORE.pod			document the CORE namespace
 lib/DBM_Filter/Changes		DBM Filter Change history
--- ../perl-5.16.0.orig/Porting/Maintainers.pl	Tue May 15 14:35:13 2012
+++ Porting/Maintainers.pl	Fri Jun  1 22:52:04 2012
@@ -2149,6 +2149,12 @@
                 ],
         'UPSTREAM' => 'blead',
     },
+    'Config_git.pl' => {
+	'MAINTAINER' => 'OpenBSD',
+	'FILES' => q[
+    		lib/Config_git.pl
+	],
+    },
 );
 
 # legacy CPAN flag
