diff -Nrux *.orig ../perl-5.16.0.orig/lib/Getopt/Std.pm ./lib/Getopt/Std.pm
--- ../perl-5.16.0.orig/lib/Getopt/Std.pm	Tue Apr 24 17:18:34 2012
+++ ./lib/Getopt/Std.pm	Sun Jun  3 22:00:37 2012
@@ -4,7 +4,7 @@
 
 =head1 NAME
 
-getopt, getopts - Process single-character switches with switch clustering
+Getopt::Std, getopt, getopts - Process single-character switches with switch clustering
 
 =head1 SYNOPSIS
 
