diff -Nru perl-5.16.0.orig/cpan/Pod-Parser/lib/Pod/Select.pm perl-5.16.0/cpan/Pod-Parser/lib/Pod/Select.pm
--- perl-5.16.0.orig/cpan/Pod-Parser/lib/Pod/Select.pm	Tue Apr 24 17:18:31 2012
+++ cpan/Pod-Parser/lib/Pod/Select.pm	Mon May 28 16:32:44 2012
@@ -18,7 +18,7 @@
 
 =head1 NAME
 
-Pod::Select, podselect() - extract selected sections of POD from input
+Pod::Select, podselect - extract selected sections of POD from input
 
 =head1 SYNOPSIS
 
