diff -Nru perl-5.16.0.orig/cpan/Pod-Parser/lib/Pod/Usage.pm perl-5.16.0/cpan/Pod-Parser/lib/Pod/Usage.pm
--- perl-5.16.0.orig/cpan/Pod-Parser/lib/Pod/Usage.pm	Tue Apr 24 17:18:31 2012
+++ cpan/Pod-Parser/lib/Pod/Usage.pm	Mon May 28 16:32:45 2012
@@ -16,7 +16,7 @@
 
 =head1 NAME
 
-Pod::Usage, pod2usage() - print a usage message from embedded pod documentation
+Pod::Usage, pod2usage - print a usage message from embedded pod documentation
 
 =head1 SYNOPSIS
 
