diff -x CVS -Nru ../perl-5.12.2/ext/DynaLoader/Makefile.PL ./ext/DynaLoader/Makefile.PL
--- ../perl-5.12.2/ext/DynaLoader/Makefile.PL	Sun Sep  5 08:14:33 2010
+++ ./ext/DynaLoader/Makefile.PL	Fri Sep 24 08:06:56 2010
@@ -55,8 +55,8 @@
 
     return "
 $object : \$(FIRST_MAKEFILE) \$(OBJECT)
-	\$(RM_RF) $object
-	\$(CP) \$(OBJECT) $object
+	#\$(RM_RF) $object
+	#\$(CP) \$(OBJECT) $object
 
 static :: $object
 ";
