diff -Nrux *.orig ../perl-5.16.0.orig/installperl ./installperl
--- ../perl-5.16.0.orig/installperl	Tue Apr 24 17:18:34 2012
+++ ./installperl	Sun Jun  3 13:47:27 2012
@@ -150,7 +150,7 @@
 }
 
 # Exclude nonxs extensions that are not architecture dependent
-my @nonxs = grep(!/^Errno$/, split(' ', $Config{'nonxs_ext'}));
+my @nonxs = grep(!/^(Errno|IO\/Compress)$/, split(' ', $Config{'nonxs_ext'}));
 
 my @ext_dirs = qw(cpan dist ext);
 foreach my $ext_dir (@ext_dirs) {
