diff -Nrux *.orig ../perl-5.16.0.orig/cpan/Module-Pluggable/t/lib/EditorJunk/Plugin/Bar.pm~ ./cpan/Module-Pluggable/t/lib/EditorJunk/Plugin/Bar.pm~
--- ../perl-5.16.0.orig/cpan/Module-Pluggable/t/lib/EditorJunk/Plugin/Bar.pm~	Tue Feb 14 21:44:41 2012
+++ ./cpan/Module-Pluggable/t/lib/EditorJunk/Plugin/Bar.pm~	Sun Jun  3 16:49:19 2012
@@ -1,9 +0,0 @@
-package EditorJunk::Bar;
-
-
-use strict;
-
-
-1;
-
-
