diff -Nrux *.orig -x *.rej ../perl-5.16.0.orig/t/porting/dual-life.t ./t/porting/dual-life.t
--- ../perl-5.16.0.orig/t/porting/dual-life.t	Tue Apr 24 17:18:36 2012
+++ ./t/porting/dual-life.t	Tue Jun 26 19:41:39 2012
@@ -21,6 +21,8 @@
   \.\./cpan/Encode/bin/u(?:cm(?:2table|lint|sort)|nidump)
    |
   \.\./cpan/Module-Build/MB-[\w\d]+/Simple/(?:test_install/)?bin/.*
+   |
+  \.\./cpan/Archive-Tar/bin/ptar.*
 )\z}ix;
 
 my %dist_dir_exe;
diff -Nrux *.orig -x *.rej ../perl-5.16.0.orig/utils/Makefile.SH ./utils/Makefile.SH
--- ../perl-5.16.0.orig/utils/Makefile.SH	Tue Apr 24 17:18:36 2012
+++ ./utils/Makefile.SH	Tue Jun 26 19:41:39 2012
@@ -48,9 +48,9 @@
 # Files to be built with variable substitution after miniperl is
 # available.  Dependencies handled manually below (for now).
 
-pl = c2ph.PL config_data.PL corelist.PL cpan.PL h2ph.PL h2xs.PL instmodsh.PL json_pp.PL perlbug.PL perldoc.PL perlivp.PL pl2pm.PL prove.PL ptar.PL ptardiff.PL ptargrep.PL cpanp-run-perl.PL cpanp.PL cpan2dist.PL shasum.PL splain.PL libnetcfg.PL piconv.PL enc2xs.PL xsubpp.PL pod2html.PL zipdetails.PL
-plextract = c2ph config_data corelist cpan h2ph h2xs instmodsh json_pp perlbug perldoc perlivp pl2pm prove ptar ptardiff ptargrep cpanp-run-perl cpanp cpan2dist shasum splain libnetcfg piconv enc2xs xsubpp pod2html zipdetails
-plextractexe = ./c2ph ./config_data ./corelist ./cpan ./h2ph ./h2xs ./json_pp ./instmodsh ./perlbug ./perldoc ./perlivp ./pl2pm ./prove ./ptar ./ptardiff ./ptargrep ./cpanp-run-perl ./cpanp ./cpan2dist ./shasum ./splain ./libnetcfg ./piconv ./enc2xs ./xsubpp ./pod2html ./zipdetails
+pl = c2ph.PL config_data.PL corelist.PL cpan.PL h2ph.PL h2xs.PL instmodsh.PL json_pp.PL perlbug.PL perldoc.PL perlivp.PL pl2pm.PL prove.PL cpanp-run-perl.PL cpanp.PL cpan2dist.PL shasum.PL splain.PL libnetcfg.PL piconv.PL enc2xs.PL xsubpp.PL pod2html.PL zipdetails.PL
+plextract = c2ph config_data corelist cpan h2ph h2xs instmodsh json_pp perlbug perldoc perlivp pl2pm prove cpanp-run-perl cpanp cpan2dist shasum splain libnetcfg piconv enc2xs xsubpp pod2html zipdetails
+plextractexe = ./c2ph ./config_data ./corelist ./cpan ./h2ph ./h2xs ./json_pp ./instmodsh ./perlbug ./perldoc ./perlivp ./pl2pm ./prove ./cpanp-run-perl ./cpanp ./cpan2dist ./shasum ./splain ./libnetcfg ./piconv ./enc2xs ./xsubpp ./pod2html ./zipdetails
 
 all: $(plextract) 
 
@@ -80,12 +80,6 @@
 perlivp:	perlivp.PL ../config.sh
 
 prove:		prove.PL ../config.sh
-
-ptar:		ptar.PL ../config.sh
-
-ptardiff:	ptardiff.PL ../config.sh
-
-ptargrep:	ptargrep.PL ../config.sh
 
 cpanp-run-perl:	cpanp-run-perl.PL ../config.sh
 
diff -Nrux *.orig -x *.rej ../perl-5.16.0.orig/utils/perlbug.PL ./utils/perlbug.PL
--- ../perl-5.16.0.orig/utils/perlbug.PL	Thu May 10 11:38:09 2012
+++ ./utils/perlbug.PL	Tue Jun 26 19:41:39 2012
@@ -344,15 +344,14 @@
 This program provides an easy way to send a thank-you message back to the
 authors and maintainers of perl.
 
-If you wish to submit a bug report, please run it without the -T flag
-(or run the program perlbug rather than perlthanks)
+If you wish to submit a bug report, please run it without the -T flag.
 EOF
 	} else {
 	    paraprint <<"EOF";
 This program provides an easy way to create a message reporting a
 bug in the core perl distribution (along with tests or patches)
 to the volunteers who maintain perl at $address.  To send a thank-you
-note to $thanksaddress instead of a bug report, please run 'perlthanks'.
+note to $thanksaddress instead of a bug report, please use the -T flag.
 
 Please do not use $0 to send test messages, test whether perl
 works, or to report bugs in perl modules from CPAN.
@@ -1170,8 +1169,6 @@
 B<perlbug> S<[ B<-v> ]> S<[ B<-r> I<returnaddress> ]>
  S<[ B<-A> ]> S<[ B<-ok> | B<-okay> | B<-nok> | B<-nokay> ]>
 
-B<perlthanks>
-
 =head1 DESCRIPTION
 
 
@@ -1324,8 +1321,8 @@
 
 =item Can you use C<perlbug> to submit a thank-you note?
 
-Yes, you can do this by either using the C<-T> option, or by invoking
-the program as C<perlthanks>. Thank-you notes are good. It makes people
+Yes, you can do this by either using the C<-T> option.
+Thank-you notes are good. It makes people
 smile. 
 
 =back
diff -Nrux *.orig -x *.rej ../perl-5.16.0.orig/utils.lst ./utils.lst
--- ../perl-5.16.0.orig/utils.lst	Tue Apr 24 17:18:36 2012
+++ ./utils.lst	Tue Jun 26 19:41:39 2012
@@ -14,16 +14,13 @@
 utils/instmodsh
 utils/json_pp
 utils/libnetcfg
-utils/perlbug	# link = utils/perlthanks
+utils/perlbug
 utils/perldoc
 utils/perlivp
 utils/piconv
 utils/pl2pm
 utils/pod2html
 utils/prove
-utils/ptar
-utils/ptardiff
-utils/ptargrep
 utils/cpanp-run-perl
 utils/cpanp
 utils/cpan2dist
