diff -Nrux *.orig ../perl-5.16.0.orig/ext/DynaLoader/Makefile.PL ./ext/DynaLoader/Makefile.PL
--- ../perl-5.16.0.orig/ext/DynaLoader/Makefile.PL	Sun May 20 06:00:24 2012
+++ ./ext/DynaLoader/Makefile.PL	Mon Jun  4 20:37:12 2012
@@ -64,8 +64,8 @@
 
     return "
 $object : \$(FIRST_MAKEFILE) \$(OBJECT)
-	\$(RM_RF) $object
-	\$(CP) \$(OBJECT) $object
+	#\$(RM_RF) $object
+	#\$(CP) \$(OBJECT) $object
 
 static :: $object
 ";
