--- ./Makefile.SH.orig	Wed May 13 13:19:28 2015
+++ ./Makefile.SH	Thu Sep 17 23:19:33 2015
@@ -719,6 +719,14 @@ case "$osname" in
 esac
 
 case "$osname" in
+openbsd)
+	$spitshell >>$Makefile <<!GROK!THIS!
+# When building Perl itself, link with the thread library,
+# so that extensions can use thread functions even though Perl
+# is not threaded
+CCDLFLAGS = $ccdlflags -lpthread
+!GROK!THIS!
+	;;
 aix)
 	$spitshell >>$Makefile <<!GROK!THIS!
 LIBS			= $perllibs
