--- ./Makefile.SH.orig	Tue Mar 18 11:53:02 2014
+++ ./Makefile.SH	Tue Mar 25 01:25:36 2014
@@ -728,6 +728,14 @@
 esac
 
 case "$osname" in
+openbsd)
+	$spitshell >>$Makefile <<!GROK!THIS!
+# When building Perl itself, link with the thread library,
+# so that extensions can use thread functions even though Perl
+# is not threaded
+CCDLFLAGS = $ccdlflags -lpthread
+!GROK!THIS!
+	;;
 aix)
 	$spitshell >>$Makefile <<!GROK!THIS!
 LIBS			= $perllibs
