--- ./Makefile.SH.orig	Mon Jan  6 14:46:42 2014
+++ ./Makefile.SH	Sat Mar 22 10:02:24 2014
@@ -695,6 +695,14 @@
 esac
 
 case "$osname" in
+openbsd)
+	$spitshell >>$Makefile <<!GROK!THIS!
+# When building Perl itself, link with the thread library,
+# so that extensions can use thread functions even though Perl
+# is not threaded
+CCDLFLAGS = $ccdlflags -lpthread
+!GROK!THIS!
+	;;
 aix)
 	$spitshell >>$Makefile <<!GROK!THIS!
 LIBS			= $perllibs
