--- ./installperl.orig	Fri Dec 18 01:58:48 2020
+++ ./installperl	Wed Jan 13 20:46:59 2021
@@ -378,7 +378,9 @@ elsif ($Is_Cygwin) { # On Cygwin symlink it to CORE to
     @corefiles = <*.h>;
 } else {
     # [als] hard-coded 'libperl' name... not good!
-    @corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    #@corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    @corefiles = <*.h *.inc perl*$Config{lib_ext}>;
+    install($libperl, "$opts{destdir}$Config{glibpth}/$libperl", "0444");
 
     # AIX needs perl.exp installed as well.
     push(@corefiles,'perl.exp') if $^O eq 'aix';
