--- ./installperl.orig	Mon May 21 05:29:23 2018
+++ ./installperl	Mon Nov 26 13:12:33 2018
@@ -377,7 +377,9 @@ elsif ($Is_Cygwin) { # On Cygwin symlink it to CORE to
     @corefiles = <*.h>;
 } else {
     # [als] hard-coded 'libperl' name... not good!
-    @corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    #@corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    @corefiles = <*.h *.inc perl*$Config{lib_ext}>;
+    install($libperl, "$opts{destdir}$Config{glibpth}/$libperl", "0444");
 
     # AIX needs perl.exp installed as well.
     push(@corefiles,'perl.exp') if $^O eq 'aix';
