--- ./installperl.orig	Wed Feb 23 18:35:27 2022
+++ ./installperl	Thu Dec  8 06:22:07 2022
@@ -339,7 +339,9 @@ elsif ($Is_Cygwin) { # On Cygwin symlink it to CORE to
     @corefiles = <*.h>;
 } else {
     # [als] hard-coded 'libperl' name... not good!
-    @corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    #@corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    @corefiles = <*.h *.inc perl*$Config{lib_ext}>;
+    install($libperl, "$opts{destdir}$Config{glibpth}/$libperl", "0444");
 
     # AIX needs perl.exp installed as well.
     push(@corefiles,'perl.exp') if $^O eq 'aix';
