--- ./installperl.orig	Wed May 13 13:19:29 2015
+++ ./installperl	Thu Sep 17 23:36:46 2015
@@ -367,7 +367,9 @@ elsif ($Is_Cygwin) { # On Cygwin symlink it to CORE to
     )
 } else {
     # [als] hard-coded 'libperl' name... not good!
-    @corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    #@corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    @corefiles = <*.h *.inc perl*$Config{lib_ext}>;
+    push(@corefiles,<libperl*.*>) unless defined($ENV{"NOLIBINSTALL"});
 
     # AIX needs perl.exp installed as well.
     push(@corefiles,'perl.exp') if $^O eq 'aix';
