--- hints/openbsd.sh.orig	Thu Dec 13 22:13:44 2012
+++ hints/openbsd.sh	Thu Dec 13 22:14:37 2012
@@ -8,12 +8,8 @@
 #	./Configure -des -Dopenbsd_distribution=defined
 #
 
-# In OpenBSD > 3.7, use perl's malloc [perl #75742]
-case "$osvers" in
-3.[89]*|[4-9]*)
-    test "$usemymalloc" || usemymalloc=y
-    ;;
-esac
+# OpenBSD has a better malloc than perl...
+test "$usemymalloc" || usemymalloc='n'
 
 # malloc wrap works
 case "$usemallocwrap" in
