--- ./ext/NDBM_File/Makefile.PL.orig	Sun Aug 11 19:35:56 2013
+++ ./ext/NDBM_File/Makefile.PL	Sun Sep 29 17:56:16 2013
@@ -1,7 +1,7 @@
 use ExtUtils::MakeMaker;
 WriteMakefile(
     NAME	=> 'NDBM_File',
-    LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
+    #LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
     XSPROTOARG => '-noprototypes', 		# XXX remove later?
     VERSION_FROM => 'NDBM_File.pm',
     INC => ($^O eq "MacOS" ? "-i ::::db:include" : "")
--- ./lib/AnyDBM_File.pm.orig	Thu May 23 16:38:10 2013
+++ ./lib/AnyDBM_File.pm	Sun Sep 29 17:56:16 2013
@@ -22,8 +22,6 @@
 
 AnyDBM_File - provide framework for multiple DBMs
 
-NDBM_File, DB_File, GDBM_File, SDBM_File, ODBM_File - various DBM implementations
-
 =head1 SYNOPSIS
 
     use AnyDBM_File;
