--- ./ext/NDBM_File/Makefile.PL.orig	Wed May 13 13:19:29 2015
+++ ./ext/NDBM_File/Makefile.PL	Thu Sep 17 23:57:40 2015
@@ -1,7 +1,7 @@
 use ExtUtils::MakeMaker;
 WriteMakefile(
     NAME	=> 'NDBM_File',
-    LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
+    #LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
     XSPROTOARG => '-noprototypes', 		# XXX remove later?
     VERSION_FROM => 'NDBM_File.pm',
     INC => ($^O eq "MacOS" ? "-i ::::db:include" : "")
--- ./lib/AnyDBM_File.pm.orig	Wed May 13 13:19:30 2015
+++ ./lib/AnyDBM_File.pm	Thu Sep 17 23:57:40 2015
@@ -22,8 +22,6 @@ __END__
 
 AnyDBM_File - provide framework for multiple DBMs
 
-NDBM_File, DB_File, GDBM_File, SDBM_File, ODBM_File - various DBM implementations
-
 =head1 SYNOPSIS
 
     use AnyDBM_File;
