--- ./Makefile.SH.orig	Sat Jul 16 03:52:42 2016
+++ ./Makefile.SH	Wed Oct 12 20:13:20 2016
@@ -1066,10 +1066,7 @@ $(CONFIGPOD): config.sh $(MINIPERL_EXE) configpm Porti
 unidatafiles $(unidatafiles) pod/perluniprops.pod: uni.data
 
 uni.data: $(MINIPERL_EXE) $(CONFIGPM) lib/unicore/mktables $(nonxs_ext)
-	$(MINIPERL) lib/unicore/mktables -C lib/unicore -P pod -maketest -makelist -p
-#	Commented out so always runs, mktables looks at far more files than we
-#	can in this makefile to decide if needs to run or not
-#	touch uni.data
+	touch uni.data
 
 # $(PERL_EXE) and ext because pod_lib.pl needs Digest::MD5
 # But also this ensures that all extensions are built before we try to scan
