--- ./MANIFEST.orig	Fri Jul 25 19:01:43 2025
+++ ./MANIFEST	Fri Jul 25 19:01:50 2025
@@ -14,6 +14,7 @@ Changes					Describe how to peruse changes between rel
 charclass_invlists.inc			Generated compiled-in inversion lists
 class.c					Internals of the `use feature 'class'` object system
 CODE_OF_CONDUCT.md			Information on where to find the Standards of Conduct
+config.over				Site-specific overrides for Configure defaults
 config_h.SH				Produces config.h
 configpm				Produces lib/Config.pm
 Configure				Portability tool
@@ -37,6 +38,7 @@ fakesdio.h				stdio in terms of PerlIO
 feature.h				Feature header
 form.h					Public declarations for formats
 generate_uudmap.c			Generate uudmap.h, the uuencode decoding map
+git_version.h				Pre-generated git_version.h for OpenBSD
 globals.c				File to declare global symbols (for shared library)
 globvar.sym				Global variables that need hiding when embedded
 gv.c					Glob value code
@@ -5376,6 +5378,7 @@ lib/Class/Struct.t			See if Class::Struct works
 lib/Config.t				See if Config works
 lib/Config/Extensions.pm		Convenient hash lookup for built extensions
 lib/Config/Extensions.t			See if Config::Extensions works
+lib/Config_git.pl			Pre-generated Config_git.pl for OpenBSD
 lib/CORE.pod				document the CORE namespace
 lib/DB.pm				Debugger API (draft)
 lib/DB.t				See if DB works
--- ./Makefile.SH.orig	Fri Jul 25 19:01:43 2025
+++ ./Makefile.SH	Fri Jul 25 19:01:47 2025
@@ -505,7 +505,7 @@ shextract=`SH_to_target $SH`
 ## In the following dollars and backticks do not need the extra backslash.
 $spitshell >>$Makefile <<!GROK!THIS!
 
-private = preplibrary \$(CONFIGPM) \$(CONFIGPOD) git_version.h lib/buildcustomize.pl
+private = preplibrary \$(CONFIGPM) \$(CONFIGPOD) lib/buildcustomize.pl
 
 # Files to be built with variable substitution before miniperl
 # is available.
@@ -695,16 +695,17 @@ $(MANIFEST_SRT):	MANIFEST $(PERL_EXE)
 
 .PHONY: all utilities
 
+# OpenBSD uses pre-generated lib/Config_git.pl and git_version.h files
 # Both git_version.h and lib/Config_git.pl are built
 # by make_patchnum.pl.
-git_version.h: lib/Config_git.pl
+#git_version.h: lib/Config_git.pl
+#
+#lib/Config_git.pl: $(MINIPERL_EXE) make_patchnum.pl
+#	$(MINIPERL) make_patchnum.pl
+#
+## make sure that we recompile perl.c if the git version changes
+#perl$(OBJ_EXT): git_version.h
 
-lib/Config_git.pl: $(MINIPERL_EXE) make_patchnum.pl
-	$(MINIPERL) make_patchnum.pl
-
-# make sure that we recompile perl.c if the git version changes
-perl$(OBJ_EXT): git_version.h
-
 !NO!SUBS!
 
 # Making utilities requires Cwd.  If we have dynamic
@@ -1376,7 +1377,7 @@ _mopup:
 	-rm -f *perl.xok
 	-rm -f cygwin.c libperl*.def libperl*.dll cygperl*.dll *.exe.stackdump
 	-rm -f $(PERL_EXE) $(MINIPERL_EXE) $(LIBPERL) libperl.*
-	-rm -f config.arch config.over $(DTRACE_H)
+	-rm -f $(DTRACE_H)
 
 _cleaner1:
 	-cd os2; rm -f Makefile
--- ./Porting/Maintainers.pl.orig	Tue Jun 24 08:23:20 2025
+++ ./Porting/Maintainers.pl	Fri Jul 25 19:01:47 2025
@@ -1509,6 +1509,9 @@ our %Modules = (
                 lib/vmsish.{pm,t}
                 ],
     },
+    'openbsd' => {
+        'FILES'      => q[lib/Config_git.pl],
+    },
 );
 
 our %DistName;
