--- ./Makefile.SH.orig	Wed May 13 13:19:28 2015
+++ ./Makefile.SH	Sun Oct 11 11:39:29 2015
@@ -1001,10 +1001,7 @@ $(CONFIGPOD): config.sh $(MINIPERL_EXE) configpm Porti
 unidatafiles $(unidatafiles) pod/perluniprops.pod: uni.data
 
 uni.data: $(MINIPERL_EXE) $(CONFIGPM) lib/unicore/mktables $(nonxs_ext)
-	$(MINIPERL) lib/unicore/mktables -C lib/unicore -P pod -maketest -makelist -p
-#	Commented out so always runs, mktables looks at far more files than we
-#	can in this makefile to decide if needs to run or not
-#	touch uni.data
+	touch uni.data
 
 # $(PERL_EXE) and ext because pod_lib.pl needs Digest::MD5
 # But also this ensures that all extensions are built before we try to scan
