--- ./ext/NDBM_File/Makefile.PL.orig	Mon Feb 18 01:59:22 2019
+++ ./ext/NDBM_File/Makefile.PL	Sat Dec 10 22:08:25 2022
@@ -1,7 +1,7 @@
 use ExtUtils::MakeMaker;
 WriteMakefile(
     NAME	=> 'NDBM_File',
-    LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
+    #LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
     XSPROTOARG => '-noprototypes', 		# XXX remove later?
     VERSION_FROM => 'NDBM_File.pm',
 );
--- ./lib/AnyDBM_File.pm.orig	Mon Feb 18 01:59:23 2019
+++ ./lib/AnyDBM_File.pm	Sat Dec 10 22:08:25 2022
@@ -22,8 +22,6 @@ __END__
 
 AnyDBM_File - provide framework for multiple DBMs
 
-NDBM_File, DB_File, GDBM_File, SDBM_File, ODBM_File - various DBM implementations
-
 =head1 SYNOPSIS
 
     use AnyDBM_File;
