--- ./Makefile.SH.orig	Wed May 13 13:19:28 2015
+++ ./Makefile.SH	Fri Oct  2 21:11:29 2015
@@ -924,10 +924,6 @@ lib/buildcustomize.pl: \$& \$(mini_obj) write_buildcus
 		else
 			$spitshell >>$Makefile <<'!NO!SUBS!'
 lib/buildcustomize.pl: $& $(mini_obj) write_buildcustomize.pl
-	-@rm -f miniperl.xok
-	$(CC) $(CLDFLAGS) -o $(MINIPERL_EXE) \
-	    $(mini_obj) $(libs)
-	$(LDLIBPTH) ./miniperl$(HOST_EXE_EXT) -w -Ilib -Idist/Exporter/lib -MExporter -e '<?>' || sh -c 'echo >&2 Failed to build miniperl.  Please run make minitest; exit 1'
 	$(MINIPERL) -f write_buildcustomize.pl
 !NO!SUBS!
 		fi
