--- ./Makefile.SH.orig	Mon Jan  6 14:46:42 2014
+++ ./Makefile.SH	Sat Mar 22 13:24:25 2014
@@ -571,13 +571,14 @@
 
 .PHONY: all translators utilities
 
+# OpenBSD uses pre-generated lib/Config_git.pl and git_version.h files
 # Both git_version.h and lib/Config_git.pl are built
 # by make_patchnum.pl.
-git_version.h: lib/Config_git.pl
+#git_version.h: lib/Config_git.pl
+#
+#lib/Config_git.pl: $(MINIPERL_EXE) make_patchnum.pl
+#	$(MINIPERL) make_patchnum.pl
 
-lib/Config_git.pl: $(MINIPERL_EXE) make_patchnum.pl
-	$(MINIPERL) make_patchnum.pl
-
 # make sure that we recompile perl.c if the git version changes
 perl$(OBJ_EXT): git_version.h
 
@@ -1277,14 +1278,13 @@
 	-rm -f perl.Addrs perl.Counts t/perl.Addrs t/perl.Counts *perl.xok
 	-rm -f cygwin.c libperl*.def libperl*.dll cygperl*.dll *.exe.stackdump
 	-rm -f $(PERL_EXE) $(MINIPERL_EXE) $(LIBPERL) libperl.* microperl
-	-rm -f config.arch config.over $(DTRACE_H) runtests
+	-rm -f $(DTRACE_H) runtests
 
 # Do not 'make _tidy' directly.
 _tidy:
 	-cd pod; $(LDLIBPTH) $(MAKE) clean
 	-cd utils; $(LDLIBPTH) $(MAKE) clean
 	-cd x2p; $(LDLIBPTH) $(MAKE) clean
-	-rm -f lib/Config_git.pl git_version.h
 	-@for x in $(DYNALOADER) $(dynamic_ext) $(static_ext) $(nonxs_ext) ; do \
 	$(MINIPERL) make_ext.pl --target=clean $$x MAKE=$(MAKE) ; \
 	done
--- ./MANIFEST.orig	Mon Jan  6 14:46:42 2014
+++ ./MANIFEST	Sat Mar 22 13:24:25 2014
@@ -7,6 +7,7 @@
 Changes			Describe how to peruse changes between releases
 charclass_invlists.h	Compiled-in inversion lists
 config_h.SH		Produces config.h
+config.over		Site-specific overrides for Configure defaults
 configpm		Produces lib/Config.pm
 Configure		Portability tool
 configure.com		Configure-equivalent for VMS
@@ -4131,6 +4132,7 @@
 form.h				Public declarations for formats
 generate_uudmap.c		Generate uudmap.h, the uuencode decoding map
 genpacksizetables.pl		Generate the size tables for pack/unpack
+git_version.h			Pre-generated git_version.h for OpenBSD
 globals.c			File to declare global symbols (for shared library)
 globvar.sym			Global variables that need hiding when embedded
 gv.c				Glob value code
@@ -4274,6 +4276,7 @@
 lib/Class/Struct.t		See if Class::Struct works
 lib/Config/Extensions.pm	Convenient hash lookup for built extensions
 lib/Config/Extensions.t		See if Config::Extensions works
+lib/Config_git.pl		Pre-generated Config_git.pl for OpenBSD
 lib/Config.t			See if Config works
 lib/CORE.pod			document the CORE namespace
 lib/DBM_Filter/Changes		DBM Filter Change history
--- ./Porting/Maintainers.pl.orig	Mon Jan  6 14:46:42 2014
+++ ./Porting/Maintainers.pl	Sat Mar 22 13:24:25 2014
@@ -14,6 +14,7 @@
 %Maintainers = (
     'abergman'  => 'Arthur Bergman <abergman@cpan.org>',
     'abigail'   => 'Abigail <abigail@abigail.be>',
+    'afresh1'   => 'Andrew Fresh <afresh1@openbsd.org>',
     'ambs'      => 'Alberto Simões <ambs@cpan.org>',
     'ams'       => 'Abhijit Menon-Sen <ams@cpan.org>',
     'andk'      => 'Andreas J. Koenig <andk@cpan.org>',
@@ -2213,6 +2214,12 @@
                 lib/vmsish.{pm,t}
                 ],
         'UPSTREAM' => 'blead',
+    },
+
+    'openbsd' => {
+        'MAINTAINER' => 'afresh1',
+        'FILES'      => q[lib/Config_git.pl],
+        'UPSTREAM'   => undef,
     },
 );
 
