--- t/op/crypt.t.orig	Sun Aug 30 19:59:49 2015
+++ t/op/crypt.t	Sun Aug 30 20:13:54 2015
@@ -35,12 +35,13 @@
 # algorithm.
 my $alg = '';       # Use default algorithm
 if ( !defined(crypt("ab", "cd")) ) {
-    $alg = '$5$';   # Use SHA-256
+    $alg = '$2b$12$12345678901234567890';   # Use Blowfish
 }
 
 SKIP: {
     skip ("VOS crypt ignores salt.", 1) if ($^O eq 'vos');
-    ok(substr(crypt("ab", $alg."cd"), 2) ne substr(crypt("ab", $alg."ce"), 2),
+    my $size = length $alg;
+    ok(substr(crypt("ab", $alg."cd"), $size) ne substr(crypt("ab", $alg."ce"), $size),
        "salt makes a difference");
 }
 
--- t/op/taint.t.orig	Sun Aug 30 20:16:16 2015
+++ t/op/taint.t	Sun Aug 30 20:16:47 2015
@@ -1976,7 +1976,7 @@
           # a different hashing algorithm.
           my $alg = '';       # Use default algorithm
           if ( !defined(crypt("ab", "cd")) ) {
-              $alg = '$5$';   # Use SHA-256
+              $alg = '$2b$12$12345678901234567890';   # Use Blowfish
           }
           my $x = crypt($_[0], $alg . $_[1]);
           $x
