--- ./hints/openbsd.sh.orig	Fri Dec 18 01:58:48 2020
+++ ./hints/openbsd.sh	Tue Jan 12 05:03:15 2021
@@ -31,6 +31,14 @@ case "$osvers" in
 	d_setruid=$undef
 esac
 
+# OpenBSD 5.5 on has 64 bit time_t
+case "$osvers" in
+[0-4].*|5.[0-4]) ;;
+*)
+	cppflags="$cppflags -DBIG_TIME"
+	;;
+esac
+
 #
 # Not all platforms support dynamic loading...
 # For the case of "$openbsd_distribution", the hints file
