--- ./hints/openbsd.sh.orig	Mon Sep 20 19:38:30 2021
+++ ./hints/openbsd.sh	Thu Dec  8 10:00:31 2022
@@ -31,6 +31,14 @@ case "$osvers" in
 	d_setruid=$undef
 esac
 
+# OpenBSD 5.5 on has 64 bit time_t
+case "$osvers" in
+[0-4].*|5.[0-4]) ;;
+*)
+	cppflags="$cppflags -DBIG_TIME"
+	;;
+esac
+
 #
 # Not all platforms support dynamic loading...
 # For the case of "$openbsd_distribution", the hints file
