--- ./dist/threads-shared/t/stress.t.orig	Tue Nov 28 03:57:28 2023
+++ ./dist/threads-shared/t/stress.t	Wed Jan  3 19:54:33 2024
@@ -83,7 +83,7 @@ use threads::shared;
         print "# Looping for $busycount iterations should take about 0.025s\n";
     }
 
-    my $TIMEOUT = 60;
+    my $TIMEOUT = 600;
 
     my $mutex = 1;
     share($mutex);
--- ./t/op/getppid.t.orig	Tue Nov 28 03:57:28 2023
+++ ./t/op/getppid.t	Wed Jan  3 19:54:33 2024
@@ -100,7 +100,7 @@ sub fork_and_retrieve {
 		}
 	    } else {
 		# Fudge it by waiting a bit more:
-		sleep 2;
+		sleep 3;
 	    }
 	    my $ppid2 = getppid();
 	    print $w "$how,$ppid1,$ppid2\n";
--- ./t/re/speed.t.orig	Tue Nov 28 03:57:28 2023
+++ ./t/re/speed.t	Wed Jan  3 19:54:33 2024
@@ -42,7 +42,7 @@ run_tests() unless caller;
 sub run_tests {
 
 
-    watchdog((($::running_as_thread && $::running_as_thread) ? 150 : 225));
+    watchdog((($::running_as_thread && $::running_as_thread) ? 150 : 540));
 
     {
         # [perl #120446]
