--- ./dist/threads-shared/t/stress.t.orig	Mon Mar  4 08:16:22 2013
+++ ./dist/threads-shared/t/stress.t	Wed Mar 13 11:57:39 2013
@@ -42,7 +42,7 @@
 {
     my $cnt = 50;
 
-    my $TIMEOUT = 60;
+    my $TIMEOUT = 600;
 
     my $mutex = 1;
     share($mutex);
--- ./t/op/threads.t.orig	Mon Mar  4 08:16:22 2013
+++ ./t/op/threads.t	Wed Mar 13 11:57:39 2013
@@ -135,7 +135,7 @@
 #
 # run-time usage of newCONSTSUB (as done by the IO boot code) wasn't
 # thread-safe - got occasional coredumps or malloc corruption
-watchdog(60, "process");
+watchdog(600, "process");
 {
     local $SIG{__WARN__} = sub {};   # Ignore any thread creation failure warnings
     my @t;
