--- ./installperl.orig	Wed Feb 23 18:35:27 2022
+++ ./installperl	Thu Dec  8 13:21:46 2022
@@ -188,7 +188,7 @@ my $installprivlib = "$opts{destdir}$Config{installpri
 my $installarchlib = "$opts{destdir}$Config{installarchlib}";
 my $installsitelib = "$opts{destdir}$Config{installsitelib}";
 my $installsitearch = "$opts{destdir}$Config{installsitearch}";
-my $installman1dir = "$opts{destdir}$Config{installman1dir}";
+my $installman1dir = "none";
 my $man1ext = $Config{man1ext};
 my $libperl = $Config{libperl};
 # Shared library and dynamic loading suffixes.
