--- ./installperl.orig	Mon May 21 05:29:23 2018
+++ ./installperl	Mon Nov 26 19:13:43 2018
@@ -199,7 +199,7 @@ my $installprivlib = "$opts{destdir}$Config{installpri
 my $installarchlib = "$opts{destdir}$Config{installarchlib}";
 my $installsitelib = "$opts{destdir}$Config{installsitelib}";
 my $installsitearch = "$opts{destdir}$Config{installsitearch}";
-my $installman1dir = "$opts{destdir}$Config{installman1dir}";
+my $installman1dir = "none";
 my $man1ext = $Config{man1ext};
 my $libperl = $Config{libperl};
 # Shared library and dynamic loading suffixes.
