--- installperl.orig	Mon Feb  1 10:03:26 2021
+++ installperl	Mon Feb  1 10:04:24 2021
@@ -307,6 +307,7 @@
 }
 elsif ($^O ne 'dos') {
     if (!$Is_NetWare) {
+	my $ver = ''; # don't install a versioned perl binary
 	install("perl$exe_ext", "$installbin/$perl_verbase$ver$exe_ext", "0755");
     }
     else {
@@ -411,7 +412,8 @@
 # Install main perl executables
 # Make links to ordinary names if installbin directory isn't current directory.
 
-if (! $versiononly && ! samepath($installbin, '.') && ($^O ne 'dos') && ! $Is_VMS && ! $Is_NetWare) {
+if (0) { # don't install a versioned perl binary
+#if (! $versiononly && ! samepath($installbin, '.') && ($^O ne 'dos') && ! $Is_VMS && ! $Is_NetWare) {
     safe_unlink("$installbin/$perl$exe_ext", "$installbin/suid$perl$exe_ext");
     if ($^O eq 'vos') {
 	# VOS doesn't support hard links, so use a symlink.
