--- ./installperl.orig	Fri Dec 23 11:09:43 2022
+++ ./installperl	Fri Dec 23 11:09:47 2022
@@ -285,6 +285,7 @@ if ($Is_VMS) {
     }
 }
 else {
+    my $ver = ''; # don't install a versioned perl binary
     install("perl$exe_ext", "$installbin/$perl_verbase$ver$exe_ext", "0755");
     `chtag -r "$installbin/$perl_verbase$ver$exe_ext"` if ($^O eq 'os390');
 }
@@ -373,7 +374,8 @@ if ($Is_W32) { #linking lib isn't made in root but in 
 # Install main perl executables
 # Make links to ordinary names if installbin directory isn't current directory.
 
-if (! $versiononly && ! samepath($installbin, '.') && ! $Is_VMS) {
+if (0) { # don't install a versioned perl binary
+#if (! $versiononly && ! samepath($installbin, '.') && ! $Is_VMS) {
     safe_unlink("$installbin/$perl$exe_ext", "$installbin/suid$perl$exe_ext");
     if ($^O eq 'vos') {
 	# VOS doesn't support hard links, so use a symlink.
