--- ./installperl.orig	Wed May 13 13:19:29 2015
+++ ./installperl	Fri Sep 18 00:49:28 2015
@@ -191,7 +191,7 @@ my $installprivlib = "$opts{destdir}$Config{installpri
 my $installarchlib = "$opts{destdir}$Config{installarchlib}";
 my $installsitelib = "$opts{destdir}$Config{installsitelib}";
 my $installsitearch = "$opts{destdir}$Config{installsitearch}";
-my $installman1dir = "$opts{destdir}$Config{installman1dir}";
+my $installman1dir = "none";
 my $man1ext = $Config{man1ext};
 my $libperl = $Config{libperl};
 # Shared library and dynamic loading suffixes.
