--- ./ext/re/t/reflags.t.orig	Sun Aug 11 19:44:48 2013
+++ ./ext/re/t/reflags.t	Fri Oct 25 21:43:19 2013
@@ -56,7 +56,7 @@
 SKIP: {
   if (
       !$Config::Config{d_setlocale}
-   || $Config::Config{ccflags} =~ /\bD?NO_LOCALE\b/
+   || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
   ) {
     skip "no locale support", 7
   }
--- ./lib/locale.t.orig	Sun Aug 11 19:44:48 2013
+++ ./lib/locale.t	Fri Oct 25 21:43:19 2013
@@ -15,7 +15,7 @@
     @INC = '../lib';
     unshift @INC, '.';
     require Config; import Config;
-    if (!$Config{d_setlocale} || $Config{ccflags} =~ /\bD?NO_LOCALE\b/) {
+    if (!$Config{d_setlocale} || $Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/) {
 	print "1..0\n";
 	exit;
     }
--- ./t/op/quotemeta.t.orig	Sun Aug 11 19:44:49 2013
+++ ./t/op/quotemeta.t	Fri Oct 25 21:43:19 2013
@@ -128,10 +128,18 @@
     is(quotemeta($char), "\\$char", "quotemeta '\\N{U+D7}' locale in UTF-8");
     is(length(quotemeta($char)), 2, "quotemeta '\\N{U+D7}' locale in UTF-8 length");
 
-    $char = "\N{U+D8}";  # Every non-ASCII Latin1 is quoted in locale.
-    utf8::upgrade($char);
-    is(quotemeta($char), "\\$char", "quotemeta '\\N{U+D8}' locale in UTF-8");
-    is(length(quotemeta($char)), 2, "quotemeta '\\N{U+D8}' locale in UTF-8 length");
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 2
+    }
+        $char = "\N{U+D8}";  # Every non-ASCII Latin1 is quoted in locale.
+        utf8::upgrade($char);
+        is(quotemeta($char), "\\$char", "quotemeta '\\N{U+D8}' locale in UTF-8");
+        is(length(quotemeta($char)), 2, "quotemeta '\\N{U+D8}' locale in UTF-8 length");
+    }
 
     is(quotemeta("\x{263a}"), "\\\x{263a}", "quotemeta locale Unicode quoted");
     is(length(quotemeta("\x{263a}")), 2, "quotemeta locale Unicode quoted length");
--- ./t/op/taint.t.orig	Sun Aug 11 19:44:49 2013
+++ ./t/op/taint.t	Fri Oct 25 21:43:19 2013
@@ -17,7 +17,7 @@
 use strict;
 use Config;
 
-plan tests => 797;
+plan tests => 813;
 
 $| = 1;
 
@@ -296,6 +296,12 @@
     is($res, 1,        "$desc: res value");
     is($one, 'a',      "$desc: \$1 value");
 
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 10 }
   SKIP: {
         skip 'No locale testing without d_setlocale', 10 if(!$Config{d_setlocale});
 
@@ -315,6 +321,7 @@
         is_tainted($one,   "$desc: \$1 tainted");
         is($res, 1,        "$desc: res value");
         is($one, 'abcd',   "$desc: \$1 value");
+    }
 
         $desc = "match /g with pattern tainted via locale";
 
@@ -339,6 +346,13 @@
     $s = 'abcd';
     ($res) = $s =~ /$TAINT(.+)/;
     $one = $1;
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 12
+    }
     isnt_tainted($s,   "$desc: s not tainted");
     is_tainted($res,   "$desc: res tainted");
     is_tainted($one,   "$desc: \$1 tainted");
@@ -356,9 +370,10 @@
     is($res, 'a',      "$desc: res value");
     is($res2,'b',      "$desc: res2 value");
     is($one, 'd',      "$desc: \$1 value");
+    }
 
   SKIP: {
-        skip 'No locale testing without d_setlocale', 12 if(!$Config{d_setlocale});
+        skip 'No locale testing without d_setlocale', 12 if(!$Config{d_setlocale}) || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/;
 
         $desc = "match with pattern tainted via locale, list cxt";
 
@@ -477,6 +492,13 @@
     is($one, 'd',      "$desc: \$1 value");
 
     $desc = "substitution /ge with pattern tainted";
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 18
+    }
 
     $s = 'abc';
     {
@@ -516,9 +538,10 @@
     is($s,  'abcd',    "$desc: s value");
     is($res, 'xyz',    "$desc: res value");
     is($one, 'abcd',   "$desc: \$1 value");
+    }
 
   SKIP: {
-        skip 'No locale testing without d_setlocale', 18 if(!$Config{d_setlocale});
+        skip 'No locale testing without d_setlocale', 18 if(!$Config{d_setlocale} || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/);
 
         $desc = "substitution with pattern tainted via locale";
 
@@ -659,6 +682,13 @@
 	$desc = "use re 'taint': match /g with string tainted";
 
 	$s = 'abcd' . $TAINT;
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 10
+    }
 	$res = $s =~ /(.)/g;
 	$one = $1;
 	is_tainted($s,     "$desc: s tainted");
@@ -666,6 +696,7 @@
 	is_tainted($one,   "$desc: \$1 tainted");
 	is($res, 1,        "$desc: res value");
 	is($one, 'a',      "$desc: \$1 value");
+    }
 
 	$desc = "use re 'taint': match with string tainted, list cxt";
 
@@ -683,6 +714,13 @@
 	$s = 'abcd' . $TAINT;
 	($res, $res2) = $s =~ /(.)/g;
 	$one = $1;
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 12
+    }
 	is_tainted($s,     "$desc: s tainted");
 	is_tainted($res,   "$desc: res tainted");
 	is_tainted($res2,  "$desc: res2 tainted");
@@ -690,6 +728,7 @@
 	is($res, 'a',      "$desc: res value");
 	is($res2,'b',      "$desc: res2 value");
 	is($one, 'd',      "$desc: \$1 value");
+    }
 
 	$desc = "use re 'taint': match with pattern tainted";
 
@@ -714,7 +753,7 @@
 	is($one, 'a',      "$desc: \$1 value");
 
   SKIP: {
-        skip 'No locale testing without d_setlocale', 10 if(!$Config{d_setlocale});
+        skip 'No locale testing without d_setlocale', 10 if(!$Config{d_setlocale} || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/);
 
         $desc = "use re 'taint': match with pattern tainted via locale";
 
@@ -775,7 +814,7 @@
 	is($one, 'd',      "$desc: \$1 value");
 
   SKIP: {
-        skip 'No locale testing without d_setlocale', 12 if(!$Config{d_setlocale});
+        skip 'No locale testing without d_setlocale', 12 if(!$Config{d_setlocale} || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/);
 
         $desc = "use re 'taint': match with pattern tainted via locale, list cxt";
 
@@ -824,6 +863,13 @@
 	is_tainted($one,   "$desc: \$1 tainted");
 	is($s,   'xyz',    "$desc: s value");
 	is($res, 1,        "$desc: res value");
+    SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 18
+    }
 	is($one, 'abcd',   "$desc: \$1 value");
 
 	$desc = "use re 'taint': substitution /g with string tainted";
@@ -849,6 +895,7 @@
 	is($s,   'abcd',   "$desc: s value");
 	is($res, 'xyz',    "$desc: res value");
 	is($one, 'abcd',   "$desc: \$1 value");
+    }
 
 	$desc = "use re 'taint': substitution /e with string tainted";
 
@@ -936,8 +983,7 @@
 	is($one, 'abcd',   "$desc: \$1 value");
 
   SKIP: {
-        skip 'No locale testing without d_setlocale', 18 if(!$Config{d_setlocale});
-
+        skip 'No locale testing without d_setlocale', 18 if(!$Config{d_setlocale} || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/);
         $desc = "use re 'taint': substitution with pattern tainted via locale";
 
         $s = 'abcd';
@@ -2184,7 +2230,13 @@
     isnt_tainted($b, "regex optimization of single char /[]/i doesn't taint");
 }
 
-{
+SKIP: {
+    if (
+        !$Config::Config{d_setlocale}
+    || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+    ) {
+        skip "no locale support", 4
+    }
     # RT 81230: tainted value during FETCH created extra ref to tied obj
 
     package P81230;
@@ -2315,7 +2367,7 @@
 }
 
 SKIP: {
-    skip 'No locale testing without d_setlocale', 4 if(!$Config{d_setlocale});
+    skip 'No locale testing without d_setlocale', 4 if(!$Config{d_setlocale} || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/);
 
     use feature 'fc';
     BEGIN {
--- ./t/re/pat.t.orig	Sun Aug 11 19:44:51 2013
+++ ./t/re/pat.t	Fri Oct 25 21:43:19 2013
@@ -509,7 +509,13 @@
         is(qr/\b\v$/,     '(?^:\b\v$)', 'qr/\b\v$/');
     }
 
-    {   # Test that charset modifier work, and are interpolated
+    SKIP: {   # Test that charset modifier work, and are interpolated
+        if (
+            !$Config::Config{d_setlocale}
+        || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+        ) {
+            skip "no locale support", 13
+        }
         is(qr/\b\v$/, '(?^:\b\v$)', 'Verify no locale, no unicode_strings gives default modifier');
         is(qr/(?l:\b\v$)/, '(?^:(?l:\b\v$))', 'Verify infix l modifier compiles');
         is(qr/(?u:\b\v$)/, '(?^:(?u:\b\v$))', 'Verify infix u modifier compiles');
--- ./t/uni/overload.t.orig	Sun Aug 11 19:44:49 2013
+++ ./t/uni/overload.t	Fri Oct 25 21:43:19 2013
@@ -101,6 +101,12 @@
         $have_setlocale++;
     }
 };
+if (
+    !$Config::Config{d_setlocale}
+  || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+) {
+    $have_setlocale = 0;
+}
 
 SKIP: {
     if (!$have_setlocale) {
--- ./t/uni/fold.t.orig	Sun Aug 11 19:44:49 2013
+++ ./t/uni/fold.t	Fri Oct 25 21:43:19 2013
@@ -428,7 +428,7 @@
         utf8::upgrade($utf8);
         is(fc($latin1), fc($utf8), "fc() gives the same results for \\x{$_} in Latin-1 and UTF-8 under unicode_strings");
         SKIP: {
-              skip 'No locale testing without d_setlocale', 2 if(!$Config{d_setlocale});
+              skip 'No locale testing without d_setlocale', 2 if(!$Config{d_setlocale}) || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/;
               BEGIN {
                   if($Config{d_setlocale}) {
                       require locale; import locale;
@@ -437,7 +437,13 @@
             is(fc($latin1), lc($latin1), "use locale; fc(qq{\\x{$_}}), lc(qq{\\x{$_}}) when qq{\\x{$_}} is in latin-1");
             is(fc($utf8), lc($utf8), "use locale; fc(qq{\\x{$_}}), lc(qq{\\x{$_}}) when qq{\\x{$_}} is in latin-1");
         }
-        {
+        SKIP: {
+            if (
+                !$Config::Config{d_setlocale}
+            || $Config::Config{ccflags} =~ /\bD?NO_LOCALE(_|\b)/
+            ) {
+                skip "no locale support", 2
+            }
             no feature 'unicode_strings';
             is(fc($latin1), lc($latin1), "under nothing, fc() for <256 is the same as lc");
         }
