--- ./hints/openbsd.sh.orig	Fri Jun 27 07:48:18 2025
+++ ./hints/openbsd.sh	Thu Jul  3 09:20:02 2025
@@ -163,6 +163,9 @@ ccflags="$ccflags -DNO_LOCALE_NUMERIC -DNO_LOCALE_COLL
 # ASCII characters which are a subset of C.UTF-8.
 ccflags="$ccflags -DLIBC_HANDLES_MISMATCHED_CTYPE"
 
+# OpenBSD's locale support is not that complete yet
+ccflags="-DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE $ccflags"
+
 # Seems that OpenBSD returns bogus values in _Thread_local variables in code in
 # shared objects, so we need to disable it. See GH #19109
 d_thread_local=undef
