--- ./hints/openbsd.sh.orig	Sun Sep 14 04:31:02 2014
+++ ./hints/openbsd.sh	Fri Dec  5 18:32:32 2014
@@ -144,8 +144,8 @@ case "$openbsd_distribution" in
 	siteprefix='/usr/local'
 	siteprefixexp='/usr/local'
 	# Ports installs non-std libs in /usr/local/lib so look there too
-	locincpth='/usr/local/include'
-	loclibpth='/usr/local/lib'
+	locincpth=''
+	loclibpth=''
 	# Link perl with shared libperl
 	if [ "$usedl" = "$define" -a -r shlib_version ]; then
 		useshrplib=true
--- ./installperl.orig	Sun Sep 14 04:31:02 2014
+++ ./installperl	Fri Dec  5 18:32:32 2014
@@ -733,6 +733,19 @@ sub installlib {
 	return unless $do_installprivlib;
     }
 
+    if ($name eq 'Config_heavy.pl') {
+        open my $ifh, '<', $_ or die $!;
+        $_ = "$_.local";
+        open my $ofh, '>', $_ or die $!;
+        while (my $l = <$ifh>) {
+            $l =~ s,^(ccflags|cppflags)[^=]*='[^']+,$& -I/usr/local/include,;
+            $l =~ s,^(ldflags|lddlflags)[^=]*='[^']+,$& -L/usr/local/lib,;
+            print $ofh $l;
+        }
+        close $ifh;
+        close $ofh;
+    }
+
     if ($Is_NetWare && !$opts{netware} && /\.(?:nlp|nlm|bs)$/) {
         # Don't copy .nlp,.nlm files, doesn't make sense on Windows and also
         # if copied will give problems when building new extensions.
