--- ./hints/openbsd.sh.orig	Wed May 13 13:19:29 2015
+++ ./hints/openbsd.sh	Fri Sep 18 01:06:42 2015
@@ -140,8 +140,8 @@ case "$openbsd_distribution" in
 	siteprefix='/usr/local'
 	siteprefixexp='/usr/local'
 	# Ports installs non-std libs in /usr/local/lib so look there too
-	locincpth='/usr/local/include'
-	loclibpth='/usr/local/lib'
+	locincpth=''
+	loclibpth=''
 	# Link perl with shared libperl
 	if [ "$usedl" = "$define" -a -r shlib_version ]; then
 		useshrplib=true
--- configpm.orig	Sat Dec 27 03:49:22 2014
+++ configpm	Sun Sep 18 10:29:26 2016
@@ -1090,6 +1090,18 @@
 }
 
 if ($orig_config_txt ne $config_txt or $orig_heavy_txt ne $heavy_txt) {
+    # During the build don't look in /usr/local for libs or includes
+    # but after, we want to let modules look there.
+    my $install_heavy_txt = $heavy_txt;
+    $install_heavy_txt =~ s,^(ccflags|cppflags)[^=]*='[^']+,$& -I/usr/local/include,gm;
+    $install_heavy_txt =~ s,^(ldflags|lddlflags)[^=]*='[^']+,$& -L/usr/local/lib,gm;
+
+    open INSTALL_CONFIG_HEAVY, ">", "$Config_heavy.install"
+         or die "Can't open install $Config_heavy: $!\n";
+    print INSTALL_CONFIG_HEAVY $install_heavy_txt;
+    close INSTALL_CONFIG_HEAVY;
+    print "updated install $Config_heavy\n";
+
     open CONFIG, ">", $Config_PM or die "Can't open $Config_PM: $!\n";
     open CONFIG_HEAVY, ">", $Config_heavy or die "Can't open $Config_heavy: $!\n";
     print CONFIG $config_txt;
--- installperl.orig	Sun Sep 18 16:59:49 2016
+++ installperl	Sun Sep 18 17:05:01 2016
@@ -664,6 +664,10 @@
 	return;
     }
 
+    # If we have different install version, install that instead
+    return if -e "$_.install";
+    $name =~ s/\.install$//;
+
     # ignore patch backups, RCS files, emacs backup & temp files and the
     # .exists files, .PL files, and test files.
     return if $name =~ m{\.orig$|\.rej$|~$|^#.+#$|,v$|^\.exists|\.PL$|\.plc$|\.t$|^test\.pl$|^dbm_filter_util\.pl$|^filter-util\.pl$|^uupacktool\.pl$|^\.gitignore$} ||
