--- ./installperl.orig	Wed May 13 13:19:29 2015
+++ ./installperl	Fri Sep 18 01:24:02 2015
@@ -245,9 +245,9 @@ else {
     }
 }
 
--f 't/rantests'		|| $Is_W32
-			|| warn "WARNING: You've never run 'make test' or",
-				" some tests failed! (Installing anyway.)\n";
+#-f 't/rantests'		|| $Is_W32
+#			|| warn "WARNING: You've never run 'make test' or",
+#				" some tests failed! (Installing anyway.)\n";
 } #if (!$Is_NetWare)
 
 # This will be used to store the packlist
