--- ./installperl.orig	Fri Jun 27 07:48:18 2025
+++ ./installperl	Thu Jul  3 11:11:39 2025
@@ -233,9 +233,9 @@ else {
     }
 }
 
--f 't/rantests'		|| $Is_W32
-			|| warn "WARNING: You've never run 'make test' or",
-				" some tests failed! (Installing anyway.)\n";
+#-f 't/rantests'		|| $Is_W32
+#			|| warn "WARNING: You've never run 'make test' or",
+#				" some tests failed! (Installing anyway.)\n";
 
 # This will be used to store the packlist
 $packlist = ExtUtils::Packlist->new("$installarchlib/.packlist");
