--- ./installperl.orig	Sun Aug 11 19:44:48 2013
+++ ./installperl	Sun Sep 29 18:55:30 2013
@@ -250,9 +250,9 @@
     }
 }
 
--f 't/rantests'		|| $Is_W32
-			|| warn "WARNING: You've never run 'make test' or",
-				" some tests failed! (Installing anyway.)\n";
+#-f 't/rantests'		|| $Is_W32
+#			|| warn "WARNING: You've never run 'make test' or",
+#				" some tests failed! (Installing anyway.)\n";
 } #if (!$Is_NetWare)
 
 # This will be used to store the packlist
