--- ./pod/perlmodinstall.pod.orig	Sun Aug 11 19:44:48 2013
+++ ./pod/perlmodinstall.pod	Mon Sep 30 07:12:23 2013
@@ -79,11 +79,11 @@
 
 Or, you can combine this step with the next to save disk space:
 
-     gzip -dc yourmodule.tar.gz | tar -xof -
+     gzip -dc yourmodule.tar.gz | tar -xf -
 
 B. UNPACK
 
-Unpack the result with C<tar -xof yourmodule.tar>
+Unpack the result with C<tar -xf yourmodule.tar>
 
 C. BUILD
 
--- ./Porting/pumpkin.pod.orig	Sun Aug 11 19:44:47 2013
+++ ./Porting/pumpkin.pod	Mon Sep 30 07:12:23 2013
@@ -540,9 +540,9 @@
 5.004_07 to 5.004_08 patch as an example.
 
 	# unpack perl5.004_07/
-	gzip -d -c perl5.004_07.tar.gz | tar -xof -
+	gzip -d -c perl5.004_07.tar.gz | tar -xf -
 	# unpack perl5.004_08/
-	gzip -d -c perl5.004_08.tar.gz | tar -xof -
+	gzip -d -c perl5.004_08.tar.gz | tar -xf -
 	makepatch perl5.004_07 perl5.004_08 > perl5.004_08.pat
 
 Makepatch will automatically generate appropriate B<rm> commands to remove
