--- ./pod/perlmodinstall.pod.orig	Mon May 21 05:29:23 2018
+++ ./pod/perlmodinstall.pod	Mon Nov 26 20:53:30 2018
@@ -79,11 +79,11 @@ You can get gzip from L<ftp://prep.ai.mit.edu/pub/gnu/
 
 Or, you can combine this step with the next to save disk space:
 
-     gzip -dc yourmodule.tar.gz | tar -xof -
+     gzip -dc yourmodule.tar.gz | tar -xf -
 
 B. UNPACK
 
-Unpack the result with C<tar -xof yourmodule.tar>
+Unpack the result with C<tar -xf yourmodule.tar>
 
 C. BUILD
 
--- ./Porting/pumpkin.pod.orig	Mon May 21 05:29:23 2018
+++ ./Porting/pumpkin.pod	Mon Nov 26 20:53:30 2018
@@ -544,9 +544,9 @@ Here's how I generate a new patch.  I'll use the hypot
 5.004_07 to 5.004_08 patch as an example.
 
 	# unpack perl5.004_07/
-	gzip -d -c perl5.004_07.tar.gz | tar -xof -
+	gzip -d -c perl5.004_07.tar.gz | tar -xf -
 	# unpack perl5.004_08/
-	gzip -d -c perl5.004_08.tar.gz | tar -xof -
+	gzip -d -c perl5.004_08.tar.gz | tar -xf -
 	makepatch perl5.004_07 perl5.004_08 > perl5.004_08.pat
 
 Makepatch will automatically generate appropriate B<rm> commands to remove
