--- ./MANIFEST.orig	Fri Dec 16 15:10:36 2022
+++ ./MANIFEST	Fri Dec 16 15:10:39 2022
@@ -1595,6 +1595,9 @@ cpan/OpenBSD-MkTemp/lib/OpenBSD/MkTemp.pm		OpenBSD::Mk
 cpan/OpenBSD-MkTemp/MkTemp.xs		OpenBSD::MkTemp
 cpan/OpenBSD-MkTemp/README		OpenBSD::MkTemp Readme
 cpan/OpenBSD-MkTemp/t/OpenBSD-MkTemp.t		OpenBSD::MkTemp test file
+cpan/OpenBSD-Pledge/lib/OpenBSD/Pledge.pm	OpenBSD::Pledge
+cpan/OpenBSD-Pledge/Pledge.xs	OpenBSD::Pledge
+cpan/OpenBSD-Pledge/t/OpenBSD-Pledge.t	OpenBSD::Pledge test file
 cpan/Params-Check/lib/Params/Check.pm	Params::Check
 cpan/Params-Check/t/01_Params-Check.t	Params::Check tests
 cpan/parent/lib/parent.pm			Establish an ISA relationship with base classes at compile time
