--- ./MANIFEST.orig	Sun Aug 11 19:44:51 2013
+++ ./MANIFEST	Fri Oct 25 20:22:18 2013
@@ -1882,6 +1882,10 @@
 cpan/Object-Accessor/t/04_Object-Accessor-lvalue.t	Object::Accessor tests
 cpan/Object-Accessor/t/05_Object-Accessor-callback.t	Object::Accessor tests
 cpan/Object-Accessor/t/06_Object-Accessor-alias.t	Object::Accessor tests
+cpan/OpenBSD-MkTemp/lib/OpenBSD/MkTemp.pm	OpenBSD::MkTemp
+cpan/OpenBSD-MkTemp/MkTemp.xs	OpenBSD::MkTemp
+cpan/OpenBSD-MkTemp/README	OpenBSD::MkTemp Readme
+cpan/OpenBSD-MkTemp/t/OpenBSD-MkTemp.t	OpenBSD::MkTemp test file
 cpan/Package-Constants/lib/Package/Constants.pm	Package::Constants
 cpan/Package-Constants/t/01_list.t		Package::Constants tests
 cpan/Params-Check/lib/Params/Check.pm	Params::Check
