--- ./Makefile.SH.orig	Tue Mar 18 11:53:02 2014
+++ ./Makefile.SH	Wed Mar 26 22:48:56 2014
@@ -1323,7 +1323,7 @@
 	-rmdir lib/TAP/Formatter lib/TAP lib/Sys/Syslog lib/Sys lib/Search
 	-rmdir lib/Scalar lib/Pod/Text lib/Pod/Simple lib/Pod/Perldoc
 	-rmdir lib/PerlIO/via lib/PerlIO lib/Perl lib/Parse/CPAN lib/Parse
-	-rmdir lib/Params lib/Package lib/Net/FTP lib/Module/Load
+	-rmdir lib/Params lib/Package lib/OpenBSD lib/Net/FTP lib/Module/Load
 	-rmdir lib/Module/CoreList lib/Module/Build/Platform lib/Module/Build
 	-rmdir lib/Module lib/Memoize lib/Math/BigInt lib/Math/BigFloat
 	-rmdir lib/Math lib/MIME lib/Locale/Maketext lib/Locale/Codes
--- ./MANIFEST.orig	Tue Mar 18 10:48:51 2014
+++ ./MANIFEST	Wed Mar 26 22:48:56 2014
@@ -1716,6 +1716,10 @@
 cpan/NEXT/t/next.t		NEXT
 cpan/NEXT/t/stringify.t		NEXT
 cpan/NEXT/t/unseen.t		NEXT
+cpan/OpenBSD-MkTemp/lib/OpenBSD/MkTemp.pm	OpenBSD::MkTemp
+cpan/OpenBSD-MkTemp/MkTemp.xs	OpenBSD::MkTemp
+cpan/OpenBSD-MkTemp/README	OpenBSD::MkTemp Readme
+cpan/OpenBSD-MkTemp/t/OpenBSD-MkTemp.t	OpenBSD::MkTemp test file
 cpan/Package-Constants/lib/Package/Constants.pm	Package::Constants
 cpan/Package-Constants/t/01_list.t		Package::Constants tests
 cpan/Params-Check/lib/Params/Check.pm	Params::Check
--- ./win32/Makefile.orig	Mon Mar  3 09:52:44 2014
+++ ./win32/Makefile	Wed Mar 26 22:48:56 2014
@@ -1252,6 +1252,7 @@
 	-if exist $(LIBDIR)\MIME rmdir /s /q $(LIBDIR)\MIME
 	-if exist $(LIBDIR)\Module rmdir /s /q $(LIBDIR)\Module
 	-if exist $(LIBDIR)\Net\FTP rmdir /s /q $(LIBDIR)\Net\FTP
+	-if exist $(LIBDIR)\OpenBSD rmdir /s /q $(LIBDIR)\OpenBSD
 	-if exist $(LIBDIR)\Package rmdir /s /q $(LIBDIR)\Package
 	-if exist $(LIBDIR)\Params rmdir /s /q $(LIBDIR)\Params
 	-if exist $(LIBDIR)\Parse rmdir /s /q $(LIBDIR)\Parse
--- ./win32/makefile.mk.orig	Mon Mar  3 09:52:44 2014
+++ ./win32/makefile.mk	Wed Mar 26 22:48:56 2014
@@ -1435,6 +1435,7 @@
 	-if exist $(LIBDIR)\MIME rmdir /s /q $(LIBDIR)\MIME
 	-if exist $(LIBDIR)\Module rmdir /s /q $(LIBDIR)\Module
 	-if exist $(LIBDIR)\Net\FTP rmdir /s /q $(LIBDIR)\Net\FTP
+	-if exist $(LIBDIR)\OpenBSD rmdir /s /q $(LIBDIR)\OpenBSD
 	-if exist $(LIBDIR)\Package rmdir /s /q $(LIBDIR)\Package
 	-if exist $(LIBDIR)\Params rmdir /s /q $(LIBDIR)\Params
 	-if exist $(LIBDIR)\Parse rmdir /s /q $(LIBDIR)\Parse
