Index: MANIFEST
===================================================================
RCS file: /data/mirror/openbsd/cvs/src/gnu/usr.bin/perl/MANIFEST,v
retrieving revision 1.62
diff -u -p -r1.62 MANIFEST
--- MANIFEST	1 Mar 2021 23:23:17 -0000	1.62
+++ MANIFEST	8 Feb 2022 22:34:21 -0000
@@ -3046,6 +3046,7 @@ dist/Devel-PPPort/devel/mktodo		Devel::P
 dist/Devel-PPPort/devel/mktodo.pl	Devel::PPPort baseline/todo generator
 dist/Devel-PPPort/devel/regenerate	Devel::PPPort API re-generator
 dist/Devel-PPPort/devel/scanprov	Devel::PPPort provided API scanner
+dist/Devel-PPPort/devel/update_release_date.pl
 dist/Devel-PPPort/HACKERS		Devel::PPPort hackers documentation
 dist/Devel-PPPort/Makefile.PL		Devel::PPPort makefile writer
 dist/Devel-PPPort/MANIFEST.SKIP		Devel::PPPort manifest skip file
@@ -3230,6 +3231,8 @@ dist/Devel-PPPort/parts/base/5027010	Dev
 dist/Devel-PPPort/parts/base/5027011	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5028000	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5028001	Devel::PPPort baseline todo file
+dist/Devel-PPPort/parts/base/5028002
+dist/Devel-PPPort/parts/base/5028003
 dist/Devel-PPPort/parts/base/5029000	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5029001	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5029002	Devel::PPPort baseline todo file
@@ -3242,6 +3245,10 @@ dist/Devel-PPPort/parts/base/5029008	Dev
 dist/Devel-PPPort/parts/base/5029009	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5029010	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5030000	Devel::PPPort baseline todo file
+dist/Devel-PPPort/parts/base/5030001
+dist/Devel-PPPort/parts/base/5030002
+dist/Devel-PPPort/parts/base/5030003
+dist/Devel-PPPort/parts/base/5031000
 dist/Devel-PPPort/parts/base/5031001	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5031002	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5031003	Devel::PPPort baseline todo file
@@ -3250,6 +3257,32 @@ dist/Devel-PPPort/parts/base/5031005	Dev
 dist/Devel-PPPort/parts/base/5031006	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5031007	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/base/5031008
+dist/Devel-PPPort/parts/base/5031009
+dist/Devel-PPPort/parts/base/5031010
+dist/Devel-PPPort/parts/base/5031011
+dist/Devel-PPPort/parts/base/5032000
+dist/Devel-PPPort/parts/base/5032001
+dist/Devel-PPPort/parts/base/5033000
+dist/Devel-PPPort/parts/base/5033001
+dist/Devel-PPPort/parts/base/5033002
+dist/Devel-PPPort/parts/base/5033003
+dist/Devel-PPPort/parts/base/5033004
+dist/Devel-PPPort/parts/base/5033005
+dist/Devel-PPPort/parts/base/5033006
+dist/Devel-PPPort/parts/base/5033007
+dist/Devel-PPPort/parts/base/5033008
+dist/Devel-PPPort/parts/base/5033009
+dist/Devel-PPPort/parts/base/5034000
+dist/Devel-PPPort/parts/base/5035001
+dist/Devel-PPPort/parts/base/5035002
+dist/Devel-PPPort/parts/base/5035003
+dist/Devel-PPPort/parts/base/5035004
+dist/Devel-PPPort/parts/base/5035005
+dist/Devel-PPPort/parts/base/5035006
+dist/Devel-PPPort/parts/base/5035007
+dist/Devel-PPPort/parts/base/5035008
+dist/Devel-PPPort/parts/base/5035009
+dist/Devel-PPPort/parts/base/5035010
 dist/Devel-PPPort/parts/embed.fnc	Devel::PPPort Perl API listing
 dist/Devel-PPPort/parts/inc/01_test	Devel::PPPort include
 dist/Devel-PPPort/parts/inc/call	Devel::PPPort include
@@ -3472,6 +3505,8 @@ dist/Devel-PPPort/parts/todo/5027010	Dev
 dist/Devel-PPPort/parts/todo/5027011	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5028000	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5028001	Devel::PPPort baseline todo file
+dist/Devel-PPPort/parts/todo/5028002
+dist/Devel-PPPort/parts/todo/5028003
 dist/Devel-PPPort/parts/todo/5029000	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5029001	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5029002	Devel::PPPort baseline todo file
@@ -3484,6 +3519,10 @@ dist/Devel-PPPort/parts/todo/5029008	Dev
 dist/Devel-PPPort/parts/todo/5029009	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5029010	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5030000	Devel::PPPort baseline todo file
+dist/Devel-PPPort/parts/todo/5030001
+dist/Devel-PPPort/parts/todo/5030002
+dist/Devel-PPPort/parts/todo/5030003
+dist/Devel-PPPort/parts/todo/5031000
 dist/Devel-PPPort/parts/todo/5031001	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5031002	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5031003	Devel::PPPort baseline todo file
@@ -3492,6 +3531,32 @@ dist/Devel-PPPort/parts/todo/5031005	Dev
 dist/Devel-PPPort/parts/todo/5031006	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5031007	Devel::PPPort baseline todo file
 dist/Devel-PPPort/parts/todo/5031008
+dist/Devel-PPPort/parts/todo/5031009
+dist/Devel-PPPort/parts/todo/5031010
+dist/Devel-PPPort/parts/todo/5031011
+dist/Devel-PPPort/parts/todo/5032000
+dist/Devel-PPPort/parts/todo/5032001
+dist/Devel-PPPort/parts/todo/5033000
+dist/Devel-PPPort/parts/todo/5033001
+dist/Devel-PPPort/parts/todo/5033002
+dist/Devel-PPPort/parts/todo/5033003
+dist/Devel-PPPort/parts/todo/5033004
+dist/Devel-PPPort/parts/todo/5033005
+dist/Devel-PPPort/parts/todo/5033006
+dist/Devel-PPPort/parts/todo/5033007
+dist/Devel-PPPort/parts/todo/5033008
+dist/Devel-PPPort/parts/todo/5033009
+dist/Devel-PPPort/parts/todo/5034000
+dist/Devel-PPPort/parts/todo/5035001
+dist/Devel-PPPort/parts/todo/5035002
+dist/Devel-PPPort/parts/todo/5035003
+dist/Devel-PPPort/parts/todo/5035004
+dist/Devel-PPPort/parts/todo/5035005
+dist/Devel-PPPort/parts/todo/5035006
+dist/Devel-PPPort/parts/todo/5035007
+dist/Devel-PPPort/parts/todo/5035008
+dist/Devel-PPPort/parts/todo/5035009
+dist/Devel-PPPort/parts/todo/5035010
 dist/Devel-PPPort/PPPort.xs		Devel::PPPort dummy PPPort.xs
 dist/Devel-PPPort/ppport_h.PL		Devel::PPPort ppport.h writer
 dist/Devel-PPPort/PPPort_pm.PL		Devel::PPPort PPPort.pm writer
