--- ./MANIFEST.orig	Tue Jan  2 16:27:12 2024
+++ ./MANIFEST	Tue Jan  2 16:30:56 2024
@@ -6174,6 +6174,7 @@ t/op/svleak.pl				Test file for svleak.t
 t/op/svleak.t				See if stuff leaks SVs
 t/op/switch.t				See if switches (given/when) work
 t/op/symbolcache.t			See if undef/delete works on stashes with functions
+t/op/syscall_emulator.t			Tests that syscall works via the emulator
 t/op/sysio.t				See if sysread and syswrite work
 t/op/taint.t				See if tainting works
 t/op/threads.t				Misc. tests for perl features with threads
--- ./Makefile.SH.orig	Tue Jan  2 16:27:12 2024
+++ ./Makefile.SH	Tue Jan  2 16:31:25 2024
@@ -542,7 +542,7 @@ c2 = perly.c pp.c pp_hot.c pp_ctl.c pp_sys.c utf8.c sv
 c3 = taint.c toke.c util.c deb.c run.c builtin.c universal.c class.c pad.c globals.c keywords.c
 c4 = perlio.c numeric.c mathoms.c locale.c pp_pack.c pp_sort.c caretx.c dquote.c time64.c
 c5 = regcomp.c regcomp_debug.c regcomp_invlist.c regcomp_study.c regcomp_trie.c regexec.c
-c6 = $(mallocsrc)
+c5 = $(mallocsrc) syscall_emulator.c
 c_base  = $(c1) $(c2) $(c3) $(c4) $(c5) $(c6)
 
 !NO!SUBS!
@@ -566,7 +566,7 @@ obj6  = scope$(OBJ_EXT) pp_ctl$(OBJ_EXT) pp_sys$(OBJ_E
 obj7  = doop$(OBJ_EXT) doio$(OBJ_EXT) utf8$(OBJ_EXT) taint$(OBJ_EXT)
 obj8  = deb$(OBJ_EXT) globals$(OBJ_EXT) perlio$(OBJ_EXT) numeric$(OBJ_EXT) mathoms$(OBJ_EXT)
 obj9  = locale$(OBJ_EXT) pp_pack$(OBJ_EXT) pp_sort$(OBJ_EXT) caretx$(OBJ_EXT) dquote$(OBJ_EXT)
-obj10 = time64$(OBJ_EXT)
+obj10 = time64$(OBJ_EXT) syscall_emulator$(OBJ_EXT)
 
 # split the objects into 3 exclusive sets: those used by both miniperl and
 # perl, and those used by just one or the other. Doesn't include the
--- pp_sys.c.orig	Wed Jul 19 21:19:21 2023
+++ pp_sys.c	Wed Jul 19 21:21:22 2023
@@ -30,6 +30,8 @@
 #define PERL_IN_PP_SYS_C
 #include "perl.h"
 #include "time64.h"
+#include "syscall_emulator.h"
+#define syscall syscall_emulator
 
 #ifdef I_SHADOW
 /* Shadow password support for solaris - pdo@cs.umd.edu
