--- ./Configure.orig	Mon May 21 05:29:23 2018
+++ ./Configure	Mon Nov 26 22:07:31 2018
@@ -20351,9 +20351,9 @@ case "$ccflags" in
 	;;
 esac
 
-randfunc=Perl_drand48
-drand01="Perl_drand48()"
-seedfunc="Perl_drand48_init"
+randfunc=drand48
+drand01="drand48()"
+seedfunc="srand48"
 randbits=48
 randseedtype=U32
 
--- ./pp.c.orig	Mon May 21 05:29:23 2018
+++ ./pp.c	Mon Nov 26 22:07:31 2018
@@ -2941,12 +2941,13 @@ PP(pp_srand)
                              "Integer overflow in srand");
             anum = UV_MAX;
         }
+        (void)srand48_deterministic((Rand_seed_t)anum);
     }
     else {
         anum = seed();
+        (void)seedDrand01((Rand_seed_t)anum);
     }
 
-    (void)seedDrand01((Rand_seed_t)anum);
     PL_srand_called = TRUE;
     if (anum)
 	XPUSHu(anum);
