--- ./Configure.orig	Sat Dec 26 07:27:04 2020
+++ ./Configure	Mon Jan 18 10:16:50 2021
@@ -20630,9 +20630,9 @@ case "$ccflags" in
 	;;
 esac
 
-randfunc=Perl_drand48
-drand01="Perl_drand48()"
-seedfunc="Perl_drand48_init"
+randfunc=drand48
+drand01="drand48()"
+seedfunc="srand48"
 randbits=48
 randseedtype=U32
 
--- ./pp.c.orig	Fri Dec 18 02:04:36 2020
+++ ./pp.c	Mon Jan 18 10:16:51 2021
@@ -2926,12 +2926,13 @@ PP(pp_srand)
                              "Integer overflow in srand");
             anum = UV_MAX;
         }
+        (void)srand48_deterministic((Rand_seed_t)anum);
     }
     else {
         anum = seed();
+        (void)seedDrand01((Rand_seed_t)anum);
     }
 
-    (void)seedDrand01((Rand_seed_t)anum);
     PL_srand_called = TRUE;
     if (anum)
 	XPUSHu(anum);
