--- ./hints/openbsd.sh.orig	Sun Oct 20 06:31:29 2013
+++ ./hints/openbsd.sh	Tue Mar 25 04:49:38 2014
@@ -147,9 +147,9 @@
 	locincpth='/usr/local/include'
 	loclibpth='/usr/local/lib'
 	# Link perl with shared libperl
-	if [ "$usedl" = "$define" -a -r shlib_version ]; then
+	if [ "$usedl" = "$define" -a -r $src/shlib_version ]; then
 		useshrplib=true
-		libperl=`. ./shlib_version; echo libperl.so.${major}.${minor}`
+		libperl=`. $src/shlib_version; echo libperl.so.${major}.${minor}`
 	fi
 	;;
 esac
