--- hints/openbsd.sh.orig	Mon Mar  3 13:34:30 2014
+++ hints/openbsd.sh	Mon Mar  3 13:48:50 2014
@@ -104,6 +104,12 @@
    ;;
 esac
 
+# OpenBSD likes threads by default in 5.2 and newer
+case "$osvers" in
+[1-4].*|5.[1234]*) ;;
+*) usethreads="y" ;;
+esac
+
 # This script UU/usethreads.cbu will get 'called-back' by Configure 
 # after it has prompted the user for whether to use threads.
 cat > UU/usethreads.cbu <<'EOCBU'
