--- ./Configure.orig	Thu May 19 17:01:22 2022
+++ ./Configure	Fri Dec  9 06:07:34 2022
@@ -21024,9 +21024,9 @@ case "$ccflags" in
 	;;
 esac
 
-randfunc=Perl_drand48
-drand01="Perl_drand48()"
-seedfunc="Perl_drand48_init"
+randfunc=drand48
+drand01="drand48()"
+seedfunc="srand48"
 randbits=48
 randseedtype=U32
 
--- ./pp.c.orig	Fri Mar 18 04:33:24 2022
+++ ./pp.c	Fri Dec  9 06:07:34 2022
@@ -2963,12 +2963,13 @@ PP(pp_srand)
                              "Integer overflow in srand");
             anum = UV_MAX;
         }
+        (void)srand48_deterministic((Rand_seed_t)anum);
     }
     else {
         anum = seed();
+        (void)seedDrand01((Rand_seed_t)anum);
     }
 
-    (void)seedDrand01((Rand_seed_t)anum);
     PL_srand_called = TRUE;
     if (anum)
         XPUSHu(anum);
