--- ./Configure.orig	Wed May 13 13:19:28 2015
+++ ./Configure	Fri Sep 18 02:33:50 2015
@@ -20345,9 +20345,9 @@ case "$ccflags" in
 	;;
 esac
 
-randfunc=Perl_drand48
-drand01="Perl_drand48()"
-seedfunc="Perl_drand48_init"
+randfunc=drand48
+drand01="drand48()"
+seedfunc="srand48"
 randbits=48
 randseedtype=U32
 
--- ./pp.c.orig	Wed May 13 13:19:30 2015
+++ ./pp.c	Fri Sep 18 02:33:50 2015
@@ -2929,9 +2929,12 @@ PP(pp_srand)
         }
     }
     else {
-        anum = seed();
+        anum = 0;
     }
 
+    if (anum)
+        (void)srand48_deterministic((Rand_seed_t)anum);
+    else
     (void)seedDrand01((Rand_seed_t)anum);
     PL_srand_called = TRUE;
     if (anum)
