--- ./t/porting/dual-life.t.orig	Wed Jan 29 07:51:46 2014
+++ ./t/porting/dual-life.t	Thu Mar 27 21:17:27 2014
@@ -23,6 +23,12 @@
 # Exceptions that are found in dual-life bin dirs but aren't
 # installed by default; some occur only during testing:
 my $not_installed = qr{^(?:
+  \.\./cpan/Archive-Tar/bin/ptar.*
+   |
+  \.\./cpan/JSON-PP/bin/json_pp
+   |
+  \.\./cpan/IO-Compress/bin/zipdetails
+   |
   \.\./cpan/Encode/bin/u(?:cm(?:2table|lint|sort)|nidump)
    |
   \.\./cpan/Module-(?:Metadata|Build)
--- ./utils/Makefile.PL.orig	Wed Jan 29 07:51:46 2014
+++ ./utils/Makefile.PL	Thu Mar 27 21:17:27 2014
@@ -35,9 +35,9 @@
 # Files to be built with variable substitution after miniperl is
 # available.  Dependencies handled manually below (for now).
 
-pl = c2ph.PL config_data.PL corelist.PL cpan.PL h2ph.PL h2xs.PL instmodsh.PL json_pp.PL perlbug.PL perldoc.PL perlivp.PL pl2pm.PL prove.PL ptar.PL ptardiff.PL ptargrep.PL shasum.PL splain.PL libnetcfg.PL piconv.PL enc2xs.PL xsubpp.PL pod2html.PL zipdetails.PL
-plextract = c2ph config_data corelist cpan h2ph h2xs instmodsh json_pp perlbug perldoc perlivp pl2pm prove ptar ptardiff ptargrep shasum splain libnetcfg piconv enc2xs xsubpp pod2html zipdetails
-plextractexe = ./c2ph ./config_data ./corelist ./cpan ./h2ph ./h2xs ./json_pp ./instmodsh ./perlbug ./perldoc ./perlivp ./pl2pm ./prove ./ptar ./ptardiff ./ptargrep ./shasum ./splain ./libnetcfg ./piconv ./enc2xs ./xsubpp ./pod2html ./zipdetails
+pl = c2ph.PL config_data.PL corelist.PL cpan.PL h2ph.PL h2xs.PL instmodsh.PL perlbug.PL perldoc.PL perlivp.PL pl2pm.PL prove.PL splain.PL libnetcfg.PL piconv.PL enc2xs.PL xsubpp.PL pod2html.PL
+plextract = c2ph config_data corelist cpan h2ph h2xs instmodsh perlbug perldoc perlivp pl2pm prove libnetcfg piconv enc2xs xsubpp pod2html
+plextractexe = ./c2ph ./config_data ./corelist ./cpan ./h2ph ./h2xs ./instmodsh ./perlbug ./perldoc ./perlivp ./pl2pm ./prove ./splain ./libnetcfg ./piconv ./enc2xs ./xsubpp ./pod2html
 
 all: $(plextract) 
 
@@ -58,8 +58,6 @@
 
 instmodsh:	instmodsh.PL ../config.sh
 
-json_pp:	json_pp.PL ../config.sh
-
 perlbug:	perlbug.PL ../config.sh ../patchlevel.h
 
 perldoc:	perldoc.PL ../config.sh
@@ -68,16 +66,8 @@
 
 prove:		prove.PL ../config.sh
 
-ptar:		ptar.PL ../config.sh
-
-ptardiff:	ptardiff.PL ../config.sh
-
-ptargrep:	ptargrep.PL ../config.sh
-
 pl2pm:		pl2pm.PL ../config.sh
 
-shasum:		shasum.PL ../config.sh
-
 splain:		splain.PL ../config.sh ../lib/diagnostics.pm
 
 libnetcfg:	libnetcfg.PL ../config.sh
@@ -87,8 +77,6 @@
 enc2xs:		enc2xs.PL ../config.sh
 
 xsubpp:		xsubpp.PL ../config.sh
-
-zipdetails:	zipdetails.PL ../config.sh
 
 pod2html:	pod2html.PL ../config.sh ../ext/Pod-Html/bin/pod2html
 
--- ./utils/perlbug.PL.orig	Mon Jan 20 06:05:57 2014
+++ ./utils/perlbug.PL	Thu Mar 27 21:17:27 2014
@@ -349,15 +349,14 @@
 This program provides an easy way to send a thank-you message back to the
 authors and maintainers of perl.
 
-If you wish to submit a bug report, please run it without the -T flag
-(or run the program perlbug rather than perlthanks)
+If you wish to submit a bug report, please run it without the -T flag.
 EOF
 	} else {
 	    paraprint <<"EOF";
 This program provides an easy way to create a message reporting a
 bug in the core perl distribution (along with tests or patches)
 to the volunteers who maintain perl at $address.  To send a thank-you
-note to $thanksaddress instead of a bug report, please run 'perlthanks'.
+note to $thanksaddress instead of a bug report, please use the -T flag.
 
 Please do not use $0 to send test messages, test whether perl
 works, or to report bugs in perl modules from CPAN.
@@ -1239,8 +1238,6 @@
 B<perlbug> S<[ B<-v> ]> S<[ B<-r> I<returnaddress> ]>
  S<[ B<-A> ]> S<[ B<-ok> | B<-okay> | B<-nok> | B<-nokay> ]>
 
-B<perlthanks>
-
 =head1 DESCRIPTION
 
 
@@ -1393,8 +1390,8 @@
 
 =item Can you use C<perlbug> to submit a thank-you note?
 
-Yes, you can do this by either using the C<-T> option, or by invoking
-the program as C<perlthanks>. Thank-you notes are good. It makes people
+Yes, you can do this by using the C<-T> option.
+Thank-you notes are good. It makes people
 smile. 
 
 =back
--- ./utils.lst.orig	Mon Aug 12 05:25:25 2013
+++ ./utils.lst	Thu Mar 27 21:17:27 2014
@@ -11,7 +11,6 @@
 utils/h2ph
 utils/h2xs
 utils/instmodsh
-utils/json_pp
 utils/libnetcfg
 utils/perlbug	# link = utils/perlthanks
 utils/perldoc
@@ -20,13 +19,8 @@
 utils/pl2pm
 utils/pod2html
 utils/prove
-utils/ptar
-utils/ptardiff
-utils/ptargrep
-utils/shasum
 utils/splain
 utils/xsubpp
-utils/zipdetails
 x2p/a2p
 x2p/find2perl
 x2p/s2p         # link = x2p/psed
