diff -Nru perl-5.16.0.orig/configpm perl-5.16.0/configpm
--- perl-5.16.0.orig/configpm	Tue Apr 24 17:18:30 2012
+++ configpm	Mon May 28 16:32:09 2012
@@ -91,6 +91,11 @@
 		   (.{1,64})\  # Split at the last convenient space
 		  /$1\n              /gx;
 
+# libpaths that should be truncated after the first path element
+my %Libpathtrunc = map {($_,$_)}
+                   qw(archlib archlibexp privlib privlibexp sitearch sitearchexp
+                      sitelib sitelibexp);
+
 # allowed opts as well as specifies default and initial values
 my %Allowed_Opts = (
     'cross'    => '', # --cross=PLATFORM - crosscompiling for PLATFORM
@@ -333,6 +338,9 @@
     }
     $in_v = $val !~ /$quote\n/;
     next if $in_v;
+
+    # XXX - should use PERLLIB_SEP, not hard-code ':'
+    $val =~ s/^([^:]+).*${quote}\w*$/$1${quote}/ if $Libpathtrunc{$name};
 
     s,/,::,g if $Extensions{$name};
 
