--- ./pod/perlop.pod.orig	Sat May 14 06:28:57 2022
+++ ./pod/perlop.pod	Tue Dec  6 18:53:06 2022
@@ -1833,7 +1833,7 @@ and although they often accept just C<"\012">, they se
 C<"\015">.  If you get in the habit of using C<"\n"> for networking,
 you may be burned some day.
 X<newline> X<line terminator> X<eol> X<end of line>
-X<\n> X<\r> X<\r\n>
+X<\r>
 
 For constructs that do interpolate, variables beginning with "C<$>"
 or "C<@>" are interpolated.  Subscripted variables such as C<$a[3]> or
