--- ./pod/perlop.pod.orig	Tue Apr 24 19:22:54 2012
+++ ./pod/perlop.pod	Wed Jun 27 09:11:56 2012
@@ -1539,7 +1539,7 @@
 C<"\015">.  If you get in the habit of using C<"\n"> for networking,
 you may be burned some day.
 X<newline> X<line terminator> X<eol> X<end of line>
-X<\n> X<\r> X<\r\n>
+X<\r>
 
 For constructs that do interpolate, variables beginning with "C<$>"
 or "C<@>" are interpolated.  Subscripted variables such as C<$a[3]> or
