--- ./pod/perlop.pod.orig	Wed May 13 13:19:30 2015
+++ ./pod/perlop.pod	Thu Sep 17 18:44:14 2015
@@ -1615,7 +1615,7 @@ and although they often accept just C<"\012">, they se
 C<"\015">.  If you get in the habit of using C<"\n"> for networking,
 you may be burned some day.
 X<newline> X<line terminator> X<eol> X<end of line>
-X<\n> X<\r> X<\r\n>
+X<\r>
 
 For constructs that do interpolate, variables beginning with "C<$>"
 or "C<@>" are interpolated.  Subscripted variables such as C<$a[3]> or
