--- ./pod/perlop.pod.orig	Sun Aug 11 19:44:48 2013
+++ ./pod/perlop.pod	Sun Sep 29 12:40:53 2013
@@ -1541,7 +1541,7 @@
 C<"\015">.  If you get in the habit of using C<"\n"> for networking,
 you may be burned some day.
 X<newline> X<line terminator> X<eol> X<end of line>
-X<\n> X<\r> X<\r\n>
+X<\r>
 
 For constructs that do interpolate, variables beginning with "C<$>"
 or "C<@>" are interpolated.  Subscripted variables such as C<$a[3]> or
