--- ./pod/perlop.pod.orig	Sat Mar 15 07:26:05 2014
+++ ./pod/perlop.pod	Mon Mar 24 19:11:04 2014
@@ -1567,7 +1567,7 @@
 C<"\015">.  If you get in the habit of using C<"\n"> for networking,
 you may be burned some day.
 X<newline> X<line terminator> X<eol> X<end of line>
-X<\n> X<\r> X<\r\n>
+X<\r>
 
 For constructs that do interpolate, variables beginning with "C<$>"
 or "C<@>" are interpolated.  Subscripted variables such as C<$a[3]> or
