--- ./t/lib/h2ph.pht.orig	Tue Apr 30 19:52:56 2013
+++ ./t/lib/h2ph.pht	Sun Sep 29 12:55:09 2013
@@ -90,10 +90,6 @@
     }
     eval("sub flim () { 0; }") unless defined(&flim);
     eval("sub flam () { 1; }") unless defined(&flam);
-    eval 'sub blli_in_use {
-        my($blli) = @_;
-	    eval q({  ($blli->{l2_proto}) ||  ($blli->{l3_proto}); });
-    }' unless defined(&blli_in_use);
     eval 'sub multiline () {"multilinestring";}' unless defined(&multiline);
 }
 1;
--- ./utils/h2ph.PL.orig	Sun Aug 11 19:21:51 2013
+++ ./utils/h2ph.PL	Sun Sep 29 12:55:09 2013
@@ -571,7 +571,7 @@
                 $in =~ s/\?\?</{/g;                         # | ??<|  {|
                 $in =~ s/\?\?>/}/g;                         # | ??>|  }|
             }
-	    if ($in =~ /^\#ifdef __LANGUAGE_PASCAL__/) {
+	    if ($in =~ s/^\#ifdef __LANGUAGE_PASCAL__//) {
 		# Tru64 disassembler.h evilness: mixed C and Pascal.
 		while (<IN>) {
 		    last if /^\#endif/;
@@ -579,8 +579,8 @@
 		$in = "";
 		next READ;
 	    }
-	    if ($in =~ /^extern inline / && # Inlined assembler.
-		$^O eq 'linux' && $file =~ m!(?:^|/)asm/[^/]+\.h$!) {
+	    # Skip inlined functions in headers
+	    if ($in =~ s/^(extern|static) (__inline__|inline) .*[^;]\s*$//) {
 		while (<IN>) {
 		    last if /^}/;
 		}
