--- ./hints/openbsd.sh.orig	Tue Apr 30 19:52:55 2013
+++ ./hints/openbsd.sh	Sun Sep 29 17:27:02 2013
@@ -8,12 +8,8 @@
 #	./Configure -des -Dopenbsd_distribution=defined
 #
 
-# In OpenBSD > 3.7, use perl's malloc [perl #75742]
-case "$osvers" in
-3.[89]*|[4-9]*)
-    test "$usemymalloc" || usemymalloc=y
-    ;;
-esac
+# OpenBSD has a better malloc than perl...
+test "$usemymalloc" || usemymalloc='n'
 
 # malloc wrap works
 case "$usemallocwrap" in
