--- installperl.orig.good	Sat May 31 16:30:25 2014
+++ installperl	Sat May 31 16:33:30 2014
@@ -278,6 +278,10 @@
     $packlist->{"$installbin/$perldll"} = { type => 'file' };
 } # if (($Is_W32 and ! $Is_NetWare) or $Is_Cygwin)
 
+# Get the install command and flags from the environment
+my @installcmd = $ENV{"INSTALL"} || "install";
+push(@installcmd, $ENV{"INSTALL_COPY"} || "-c");
+
 # First we install the version-numbered executables.
 
 if ($Is_VMS) {
@@ -301,10 +305,8 @@
 }
 elsif ($^O ne 'dos') {
     if (!$Is_NetWare) {
-	safe_unlink("$installbin/$perl_verbase$ver$exe_ext");
-	copy("perl$exe_ext", "$installbin/$perl_verbase$ver$exe_ext");
-	strip("$installbin/$perl_verbase$ver$exe_ext");
-	chmod(0755, "$installbin/$perl_verbase$ver$exe_ext");
+	install("perl$exe_ext", "$installbin/$perl_verbase$ver$exe_ext",
+	    "0755", $dostrip);
     }
     else {
 	# If installing onto a NetWare server
@@ -650,6 +652,24 @@
 	    unless -f $to and (chmod(0666, $to), unlink $to)
 		   and File::Copy::copy($from, $to);
     }
+    $packlist->{$xto} = { type => 'file' };
+}
+
+sub install {
+    my($from,$to,$mode,$strip) = @_;
+
+    my @cmd = @installcmd;
+    push @cmd, ( '-m', $mode ) if $mode;
+    push @cmd, ( '-s'        ) if $strip;
+    push @cmd, ( $from, $to  );
+
+    my $xto = $to;
+    $xto =~ s/^\Q$opts{destdir}\E// if $opts{destdir};
+    print $opts{verbose} ? "  install $from $xto\n" : "  $xto\n"
+        unless $opts{silent};
+
+    system(@cmd);
+    warn "Couldn't @cmd\n" if $?;
     $packlist->{$xto} = { type => 'file' };
 }
 
