--- ./installperl.orig	Wed Feb 23 18:35:27 2022
+++ ./installperl	Wed Dec  7 08:33:09 2022
@@ -142,7 +142,7 @@ if ((-e "testcompile") && (defined($ENV{'COMPILE'}))) 
 }
 
 # Exclude nonxs extensions that are not architecture dependent
-my @nonxs = grep(!/^Errno$/, split(' ', $Config{'nonxs_ext'}));
+my @nonxs = grep(!/^(Errno|IO\/Compress)$/, split(' ', $Config{'nonxs_ext'}));
 
 my @ext_dirs = qw(cpan dist ext);
 foreach my $ext_dir (@ext_dirs) {
