--- ./installperl.orig	Wed May 13 13:19:29 2015
+++ ./installperl	Thu Sep 17 19:35:47 2015
@@ -145,7 +145,7 @@ if ((-e "testcompile") && (defined($ENV{'COMPILE'}))) 
 }
 
 # Exclude nonxs extensions that are not architecture dependent
-my @nonxs = grep(!/^Errno$/, split(' ', $Config{'nonxs_ext'}));
+my @nonxs = grep(!/^(Errno|IO\/Compress)$/, split(' ', $Config{'nonxs_ext'}));
 
 my @ext_dirs = qw(cpan dist ext);
 foreach my $ext_dir (@ext_dirs) {
