--- ./installperl.orig	Mon May 21 05:29:23 2018
+++ ./installperl	Mon Nov 26 09:06:26 2018
@@ -153,7 +153,7 @@ if ((-e "testcompile") && (defined($ENV{'COMPILE'}))) 
 }
 
 # Exclude nonxs extensions that are not architecture dependent
-my @nonxs = grep(!/^Errno$/, split(' ', $Config{'nonxs_ext'}));
+my @nonxs = grep(!/^(Errno|IO\/Compress)$/, split(' ', $Config{'nonxs_ext'}));
 
 my @ext_dirs = qw(cpan dist ext);
 foreach my $ext_dir (@ext_dirs) {
