--- ./Configure.orig	Tue Apr 24 17:18:30 2012
+++ ./Configure	Wed Jun 27 16:46:57 2012
@@ -5094,6 +5094,25 @@
 		eval $checkccflag
 		;;
 	esac
+	case "$gccversion" in
+	1*) ;;
+	2*) ;;
+	?*) 	echo " "
+		echo "Checking if your compiler accepts -fno-delete-null-pointer-checks" 2>&1
+		echo 'int main(void) { return 0; }' > gcctest.c
+		if $cc -O2 -fno-delete-null-pointer-checks -o gcctest gcctest.c; then
+			echo "Yes, it does." 2>&1
+			case "$ccflags" in
+			*delete-null-pointer-checks*)
+				echo "Leaving current flags $ccflags alone." 2>&1
+				;;
+			*) dflt="$dflt -fno-delete-null-pointer-checks" ;;
+			esac
+		else
+			echo "Nope, it doesn't, but that's ok." 2>&1
+		fi
+		;;
+	esac
 	# For gcc, adding -pipe speeds up compilations for some, but apparently
 	# some assemblers can't read from stdin.  (It also slows down compilations
 	# in other cases, but those are apparently rarer these days.)  AD 5/2004.
