--- Configure.orig	Tue Mar 25 22:20:16 2014
+++ Configure	Tue Mar 25 22:21:04 2014
@@ -22384,8 +22384,8 @@
 		Scalar/List/Utils) this_ext="List/Util"	;;
 		PathTools)         this_ext="Cwd"	;;
 	    esac;
-	    echo " $xs_extensions $nonxs_extensions" > $$.tmp;
-	    if $contains " $this_ext " $$.tmp; then
+	    echo " $xs_extensions $nonxs_extensions" > $tdir/$$.tmp;
+	    if $contains " $this_ext " $tdir/$$.tmp; then
 		echo >&4;
 		echo "Duplicate directories detected for extension $xxx" >&4;
 		echo "Configure cannot correctly recover from this - shall I abort?" >&4;
@@ -22403,15 +22403,15 @@
 		esac;
 		echo "Ok.  You will need to correct config.sh before running make." >&4;
 	    fi;
-	    $ls -1 $xxx > $$.tmp;
-	    if   $contains "\.xs$" $$.tmp > /dev/null 2>&1; then
+	    $ls -1 $xxx > $tdir/$$.tmp;
+	    if   $contains "\.xs$" $tdir/$$.tmp > /dev/null 2>&1; then
 		xs_extensions="$xs_extensions $this_ext";
-	    elif $contains "\.c$"  $$.tmp > /dev/null 2>&1; then
+	    elif $contains "\.c$"  $tdir/$$.tmp > /dev/null 2>&1; then
 		xs_extensions="$xs_extensions $this_ext";
 	    elif $test -d $xxx; then
 		nonxs_extensions="$nonxs_extensions $this_ext";
 	    fi;
-	    $rm -f $$.tmp;
+	    $rm -f $tdir/$$.tmp;
 	    ;;
 	esac;
     done'
