--- ./cpan/podlators/lib/Pod/Man.pm.orig	Tue Jun 24 08:23:20 2025
+++ ./cpan/podlators/lib/Pod/Man.pm	Sat Jul 26 12:23:29 2025
@@ -947,6 +947,8 @@ sub devise_title {
                     $cut = $i + 1;
                     $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                     last;
+                } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                    $cut = $i + 1;
                 }
             }
             if ($cut > 0) {
