--- cpan/podlators/lib/Pod/Man.pm.orig	Wed Apr  6 05:31:16 2016
+++ cpan/podlators/lib/Pod/Man.pm	Wed Aug 31 04:25:42 2016
@@ -896,6 +896,8 @@
                     $cut = $i + 1;
                     $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                     last;
+                } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                    $cut = $i + 1;
                 }
             }
             if ($cut > 0) {
