--- ./cpan/podlators/lib/Pod/Man.pm.orig	Tue Nov 28 03:57:28 2023
+++ ./cpan/podlators/lib/Pod/Man.pm	Sun Dec 24 17:15:14 2023
@@ -970,6 +970,8 @@ sub devise_title {
                     $cut = $i + 1;
                     $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                     last;
+                } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                    $cut = $i + 1;
                 }
             }
             if ($cut > 0) {
