--- ./cpan/podlators/lib/Pod/Man.pm.orig	Mon Dec 28 08:57:43 2020
+++ ./cpan/podlators/lib/Pod/Man.pm	Wed Dec  7 15:15:44 2022
@@ -900,6 +900,8 @@ sub devise_title {
                     $cut = $i + 1;
                     $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                     last;
+                } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                    $cut = $i + 1;
                 }
             }
             if ($cut > 0) {
