--- ./cpan/Text-Balanced/lib/Text/Balanced.pm.orig	Fri Jul  2 11:28:35 2010
+++ ./cpan/Text-Balanced/lib/Text/Balanced.pm	Mon Mar 24 23:04:22 2014
@@ -1508,7 +1508,7 @@
 If the string is "PARA", C<extract_tagged> returns only the first paragraph
 after the tag (up to the first line that is either empty or contains
 only whitespace characters).
-If the string is "", the the default behaviour (i.e. failure) is reinstated.
+If the string is "", the default behaviour (i.e. failure) is reinstated.
 
 For example, suppose the start tag "/para" introduces a paragraph, which then
 continues until the next "/endpara" tag or until another "/para" tag is
