--- ./cpan/podlators/lib/Pod/Man.pm.orig	Wed May 13 13:19:29 2015
+++ ./cpan/podlators/lib/Pod/Man.pm	Thu Sep 17 21:01:49 2015
@@ -857,7 +857,9 @@ sub devise_title {
                 $cut = $i + 1;
                 $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                 last;
-            }
+            } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                $cut = $i + 1;
+	    }
         }
         if ($cut > 0) {
             splice (@dirs, 0, $cut);
