--- ./cpan/Text-Balanced/lib/Text/Balanced.pm.orig	Wed May 13 13:19:29 2015
+++ ./cpan/Text-Balanced/lib/Text/Balanced.pm	Thu Sep 17 20:44:33 2015
@@ -1508,7 +1508,7 @@ C<extract_tagged> returns the complete text up to the 
 If the string is "PARA", C<extract_tagged> returns only the first paragraph
 after the tag (up to the first line that is either empty or contains
 only whitespace characters).
-If the string is "", the the default behaviour (i.e. failure) is reinstated.
+If the string is "", the default behaviour (i.e. failure) is reinstated.
 
 For example, suppose the start tag "/para" introduces a paragraph, which then
 continues until the next "/endpara" tag or until another "/para" tag is
