--- ./cpan/podlators/lib/Pod/Man.pm.orig	Sun Oct 20 06:31:28 2013
+++ ./cpan/podlators/lib/Pod/Man.pm	Mon Mar 24 23:20:07 2014
@@ -857,7 +857,9 @@
                 $cut = $i + 1;
                 $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                 last;
-            }
+            } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                $cut = $i + 1;
+	    }
         }
         if ($cut > 0) {
             splice (@dirs, 0, $cut);
