--- ./cpan/podlators/lib/Pod/Man.pm.orig	Sun Aug 11 19:44:48 2013
+++ ./cpan/podlators/lib/Pod/Man.pm	Sun Sep 29 15:01:37 2013
@@ -864,7 +864,9 @@
                 $cut = $i + 1;
                 $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                 last;
-            }
+            } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                $cut = $i + 1;
+	    }
         }
         if ($cut > 0) {
             splice (@dirs, 0, $cut);
