--- ./cpan/Pod-Perldoc/lib/Pod/Perldoc/ToMan.pm.orig	Sun Aug 11 19:44:48 2013
+++ ./cpan/Pod-Perldoc/lib/Pod/Perldoc/ToMan.pm	Sat Nov  9 14:45:32 2013
@@ -263,7 +263,7 @@
 	   if( $self->_is_nroff  )             { qw()              }
 	elsif( $self->_have_groff_with_utf8 )  { qw(-Kutf8 -Tutf8) }
 	elsif( $self->_is_ebcdic )             { qw(-Tcp1047)      }
-	elsif( $self->_have_mandoc_with_utf8 ) { qw(-Tutf8)        }
+	elsif( $self->_have_mandoc_with_utf8 ) { qw(-Tlocale)      }
 	elsif( $self->_is_mandoc )             { qw()              }
 	else                                   { qw(-Tlatin1)      }
 	}
