--- ./cpan/podlators/lib/Pod/Man.pm.orig	Fri Dec 18 01:58:27 2020
+++ ./cpan/podlators/lib/Pod/Man.pm	Mon Jan 11 09:43:31 2021
@@ -900,6 +900,8 @@ sub devise_title {
                     $cut = $i + 1;
                     $cut++ if ($dirs[$i + 1] && $dirs[$i + 1] eq 'lib');
                     last;
+                } elsif ($dirs[$i] eq 'lib' && $dirs[$i + 1] && $dirs[0] eq 'ext') {
+                    $cut = $i + 1;
                 }
             }
             if ($cut > 0) {
