--- ./deb.c.orig	Wed Aug 29 11:02:03 2012
+++ ./deb.c	Mon Mar 24 23:35:55 2014
@@ -234,7 +234,7 @@
 		PerlIO_printf(Perl_debug_log, "\n");
 	    else {
 
-		/* Find the the current context's stack range by searching
+		/* Find the current context's stack range by searching
 		 * forward for any higher contexts using this stack; failing
 		 * that, it will be equal to the size of the stack for old
 		 * stacks, or PL_stack_sp for the current stack
