--- ./deb.c.orig	Sun Aug 11 19:44:48 2013
+++ ./deb.c	Sun Sep 29 13:09:10 2013
@@ -234,7 +234,7 @@
 		PerlIO_printf(Perl_debug_log, "\n");
 	    else {
 
-		/* Find the the current context's stack range by searching
+		/* Find the current context's stack range by searching
 		 * forward for any higher contexts using this stack; failing
 		 * that, it will be equal to the size of the stack for old
 		 * stacks, or PL_stack_sp for the current stack
