--- ./deb.c.orig	Tue Feb 14 21:44:44 2012
+++ ./deb.c	Wed Jun 27 14:36:35 2012
@@ -234,7 +234,7 @@
 		PerlIO_printf(Perl_debug_log, "\n");
 	    else {
 
-		/* Find the the current context's stack range by searching
+		/* Find the current context's stack range by searching
 		 * forward for any higher contexts using this stack; failing
 		 * that, it will be equal to the size of the stack for old
 		 * stacks, or PL_stack_sp for the current stack
