--- ./embedvar.h.orig	Sat Feb 13 12:39:20 2021
+++ ./embedvar.h	Sat Feb 13 12:39:26 2021
@@ -401,12 +401,12 @@
 #define PL_Gfold_locale		(my_vars->Gfold_locale)
 #define PL_hash_chars		(my_vars->Ghash_chars)
 #define PL_Ghash_chars		(my_vars->Ghash_chars)
-#define PL_hash_seed		(my_vars->Ghash_seed)
-#define PL_Ghash_seed		(my_vars->Ghash_seed)
 #define PL_hash_seed_set	(my_vars->Ghash_seed_set)
 #define PL_Ghash_seed_set	(my_vars->Ghash_seed_set)
-#define PL_hash_state		(my_vars->Ghash_state)
-#define PL_Ghash_state		(my_vars->Ghash_state)
+#define PL_hash_seed_w		(my_vars->Ghash_seed_w)
+#define PL_Ghash_seed_w		(my_vars->Ghash_seed_w)
+#define PL_hash_state_w		(my_vars->Ghash_state_w)
+#define PL_Ghash_state_w	(my_vars->Ghash_state_w)
 #define PL_hints_mutex		(my_vars->Ghints_mutex)
 #define PL_Ghints_mutex		(my_vars->Ghints_mutex)
 #define PL_in_some_fold		(my_vars->Gin_some_fold)
--- ./perlapi.h.orig	Sat Feb 13 12:39:22 2021
+++ ./perlapi.h	Sat Feb 13 12:39:26 2021
@@ -153,12 +153,12 @@ END_EXTERN_C
 #define PL_fold_locale		(*Perl_Gfold_locale_ptr(NULL))
 #undef  PL_hash_chars
 #define PL_hash_chars		(*Perl_Ghash_chars_ptr(NULL))
-#undef  PL_hash_seed
-#define PL_hash_seed		(*Perl_Ghash_seed_ptr(NULL))
 #undef  PL_hash_seed_set
 #define PL_hash_seed_set	(*Perl_Ghash_seed_set_ptr(NULL))
-#undef  PL_hash_state
-#define PL_hash_state		(*Perl_Ghash_state_ptr(NULL))
+#undef  PL_hash_seed_w
+#define PL_hash_seed_w		(*Perl_Ghash_seed_w_ptr(NULL))
+#undef  PL_hash_state_w
+#define PL_hash_state_w		(*Perl_Ghash_state_w_ptr(NULL))
 #undef  PL_hints_mutex
 #define PL_hints_mutex		(*Perl_Ghints_mutex_ptr(NULL))
 #undef  PL_in_some_fold
