--- ./dist/Math-BigInt/lib/Math/BigInt.pm.orig	Sun Oct 20 06:31:29 2013
+++ ./dist/Math-BigInt/lib/Math/BigInt.pm	Tue Mar 25 00:07:12 2014
@@ -2677,11 +2677,12 @@
             next;
         }
 
-        # Don't do anything with undefs.
-
-        unless (defined($a[$i])) {
-            next;
-        }
+#        # Don't do anything with undefs.
+#
+#        unless (defined($a[$i])) {
+#            next;
+#        }
+        $a[$i] //= 0;
 
         # Perl scalars are fed to the appropriate constructor.
 
