--- ./dist/Math-BigInt/lib/Math/BigInt.pm.orig	Sun Aug 11 19:44:48 2013
+++ ./dist/Math-BigInt/lib/Math/BigInt.pm	Thu Oct 31 09:05:27 2013
@@ -2670,11 +2670,12 @@
             next;
         }
 
-        # Don't do anything with undefs.
-
-        unless (defined($a[$i])) {
-            next;
-        }
+#        # Don't do anything with undefs.
+#
+#        unless (defined($a[$i])) {
+#            next;
+#        }
+        $a[$i] //= 0;
 
         # Perl scalars are fed to the appropriate constructor.
 
