--- ./Configure.orig	Tue Nov 28 03:57:27 2023
+++ ./Configure	Sun Dec 24 20:50:27 2023
@@ -24329,6 +24329,8 @@ esac
 nonxs_ext=''
 for xxx in $nonxs_extensions ; do
 	case "$xxx" in
+	CVS|RCS|SCCS|.svn)
+		;;
 	VMS*)
 		;;
 	*)	nonxs_ext="$nonxs_ext $xxx"
--- ./cpan/Compress-Raw-Bzip2/private/MakeUtil.pm.orig	Tue Nov 28 03:57:27 2023
+++ ./cpan/Compress-Raw-Bzip2/private/MakeUtil.pm	Sun Dec 24 20:50:27 2023
@@ -35,7 +35,8 @@ sub MY::libscan
     my $path = shift;
 
     return undef
-        if $path =~ /(~|\.bak|_bak)$/ ||
+        if $path =~ /^(?:RCS|CVS|SCCS|\.svn|_darcs)$/ ||
+           $path =~ /(~|\.bak|_bak)$/ ||
            $path =~ /\..*\.sw(o|p)$/  ||
            $path =~ /\B\.svn\b/;
 
--- ./cpan/Compress-Raw-Zlib/private/MakeUtil.pm.orig	Tue Nov 28 03:57:27 2023
+++ ./cpan/Compress-Raw-Zlib/private/MakeUtil.pm	Sun Dec 24 20:50:27 2023
@@ -35,7 +35,8 @@ sub MY::libscan
     my $path = shift;
 
     return undef
-        if $path =~ /(~|\.bak|_bak)$/ ||
+        if $path =~ /^(?:RCS|CVS|SCCS|\.svn|_darcs)$/ ||
+           $path =~ /(~|\.bak|_bak)$/ ||
            $path =~ /\..*\.sw(o|p)$/  ||
            $path =~ /\B\.svn\b/;
 
--- ./cpan/IO-Compress/private/MakeUtil.pm.orig	Tue Nov 28 03:57:27 2023
+++ ./cpan/IO-Compress/private/MakeUtil.pm	Sun Dec 24 20:50:27 2023
@@ -35,7 +35,8 @@ sub MY::libscan
     my $path = shift;
 
     return undef
-        if $path =~ /(~|\.bak|_bak)$/ ||
+        if $path =~ /^(?:RCS|CVS|SCCS|\.svn|_darcs)$/ ||
+           $path =~ /(~|\.bak|_bak)$/ ||
            $path =~ /\..*\.sw(o|p)$/  ||
            $path =~ /\B\.svn\b/;
 
