--- ./Configure.orig	Fri Jun 27 07:48:18 2025
+++ ./Configure	Wed Jul  2 05:12:31 2025
@@ -24582,6 +24582,8 @@ esac
 nonxs_ext=''
 for xxx in $nonxs_extensions ; do
 	case "$xxx" in
+	CVS|RCS|SCCS|.svn)
+		;;
 	VMS*)
 		;;
 	*)	nonxs_ext="$nonxs_ext $xxx"
--- ./cpan/Compress-Raw-Bzip2/private/MakeUtil.pm.orig	Fri Jun 27 07:48:18 2025
+++ ./cpan/Compress-Raw-Bzip2/private/MakeUtil.pm	Wed Jul  2 05:12:31 2025
@@ -35,7 +35,8 @@ sub MY::libscan
     my $path = shift;
 
     return undef
-        if $path =~ /(~|\.bak|_bak)$/ ||
+        if $path =~ /^(?:RCS|CVS|SCCS|\.svn|_darcs)$/ ||
+           $path =~ /(~|\.bak|_bak)$/ ||
            $path =~ /\..*\.sw(o|p)$/  ||
            $path =~ /\B\.svn\b/;
 
--- ./cpan/Compress-Raw-Zlib/private/MakeUtil.pm.orig	Fri Jun 27 07:48:18 2025
+++ ./cpan/Compress-Raw-Zlib/private/MakeUtil.pm	Wed Jul  2 05:12:31 2025
@@ -35,7 +35,8 @@ sub MY::libscan
     my $path = shift;
 
     return undef
-        if $path =~ /(~|\.bak|_bak)$/ ||
+        if $path =~ /^(?:RCS|CVS|SCCS|\.svn|_darcs)$/ ||
+           $path =~ /(~|\.bak|_bak)$/ ||
            $path =~ /\..*\.sw(o|p)$/  ||
            $path =~ /\B\.svn\b/;
 
--- ./cpan/IO-Compress/private/MakeUtil.pm.orig	Fri Jun 27 07:48:18 2025
+++ ./cpan/IO-Compress/private/MakeUtil.pm	Wed Jul  2 05:12:31 2025
@@ -35,7 +35,8 @@ sub MY::libscan
     my $path = shift;
 
     return undef
-        if $path =~ /(~|\.bak|_bak)$/ ||
+        if $path =~ /^(?:RCS|CVS|SCCS|\.svn|_darcs)$/ ||
+           $path =~ /(~|\.bak|_bak)$/ ||
            $path =~ /\..*\.sw(o|p)$/  ||
            $path =~ /\B\.svn\b/;
 
