--- ./lib/perl5db.pl.orig	Sun Aug 11 19:44:48 2013
+++ ./lib/perl5db.pl	Sun Sep 29 15:59:28 2013
@@ -2969,7 +2969,7 @@
 =head4 C<$rc> - Recall command
 
 Manages the commands in C<@hist> (which is created if C<Term::ReadLine> reports
-that the terminal supports history). It find the the command required, puts it
+that the terminal supports history). It find the command required, puts it
 into C<$cmd>, and redoes the loop to execute it.
 
 =cut
