--- ./config_h.SH.orig	Mon Jan  6 14:46:42 2014
+++ ./config_h.SH	Wed Jul  9 16:31:44 2014
@@ -261,6 +261,12 @@
  */
 #$d_locconv HAS_LOCALECONV	/**/
 
+/* HAS_LC_MONETARY_2008:
+ *	This symbol, if defined, indicates that the localeconv routine is
+ *	available and has the additional members added in POSIX 1003.1-2008.
+ */
+#$d_lc_monetary_2008 HAS_LC_MONETARY_2008	/**/
+
 /* HAS_LOCKF:
  *	This symbol, if defined, indicates that the lockf routine is
  *	available to do file locking.
--- ./Configure.orig	Mon Jan  6 14:46:42 2014
+++ ./Configure	Wed Jul  9 16:31:44 2014
@@ -580,6 +580,7 @@
 d_localtime_r_needs_tzset=''
 localtime_r_proto=''
 d_locconv=''
+d_lc_monetary_2008=''
 d_lockf=''
 d_longdbl=''
 longdblsize=''
@@ -15383,6 +15384,27 @@
 set localeconv d_locconv
 eval $inlibc
 
+: see if localtime_r calls tzset
+case "$d_locconv:$d_lc_monetary_2008" in
+define:)
+	$cat >try.c <<EOCP
+#include <locale.h>
+int main() {
+	struct lconv *lc = localeconv();
+	char int_p_cs_precedes = lc->int_p_cs_precedes;
+	return 0;
+}
+EOCP
+	set try
+	if eval $compile; then
+		d_lc_monetary_2008="$define"
+	else
+		d_lc_monetary_2008="$undef"
+	fi;
+	$rm_try
+	;;
+esac
+
 : see if lockf exists
 set lockf d_lockf
 eval $inlibc
@@ -23172,6 +23194,7 @@
 d_localtime_r='$d_localtime_r'
 d_localtime_r_needs_tzset='$d_localtime_r_needs_tzset'
 d_locconv='$d_locconv'
+d_lc_monetary_2008='$d_lc_monetary_2008'
 d_lockf='$d_lockf'
 d_longdbl='$d_longdbl'
 d_longlong='$d_longlong'
--- ./configure.com.orig	Mon Jan  6 14:46:42 2014
+++ ./configure.com	Wed Jul  9 16:31:44 2014
@@ -5447,6 +5447,7 @@
 $   i_locale="define"
 $   i_langinfo="define"
 $   d_locconv="define"
+$   d_lc_monetary_2008="define"
 $   d_nl_langinfo="define"
 $   d_setlocale="define"
 $   vms_cc_type="decc"
@@ -5467,6 +5468,7 @@
 $   i_locale="undef"
 $   i_langinfo="undef"
 $   d_locconv="undef"
+$   d_lc_monetary_2008="undef"
 $   d_nl_langinfo="undef"
 $   d_setlocale="undef"
 $ ENDIF
@@ -6116,6 +6118,7 @@
 $ WC "d_llseek='undef'"
 $ WC "d_localtime64='undef'"
 $ WC "d_locconv='" + d_locconv + "'"
+$ WC "d_lc_monetary_2008='" + d_lc_monetary_2008 + "'"
 $ WC "d_lockf='undef'"
 $ WC "d_longdbl='" + d_longdbl + "'"
 $ WC "d_longlong='" + d_longlong + "'"
--- ./Cross/config.sh-arm-linux.orig	Mon Jan  6 14:46:42 2014
+++ ./Cross/config.sh-arm-linux	Wed Jul  9 16:31:44 2014
@@ -296,6 +296,7 @@
 d_isnan='define'
 d_isnanl='define'
 d_killpg='define'
+d_lc_monetary_2008='undef'
 d_lchown='define'
 d_ldbl_dig='define'
 d_libm_lib_version='define'
--- ./ext/POSIX/lib/POSIX.pod.orig	Mon Jan  6 14:46:45 2014
+++ ./ext/POSIX/lib/POSIX.pod	Wed Jul  9 16:31:44 2014
@@ -752,6 +752,12 @@
 		n_sep_by_space
 		p_sign_posn
 		n_sign_posn
+		int_p_cs_precedes
+		int_p_sep_by_space
+		int_n_cs_precedes
+		int_n_sep_by_space
+		int_p_sign_posn
+		int_n_sign_posn
 	))
 	{
 		printf qq(%s: "%s",\n), $property, $lconv->{$property};
--- ./ext/POSIX/POSIX.xs.orig	Mon Jan  6 14:46:45 2014
+++ ./ext/POSIX/POSIX.xs	Wed Jul  9 16:31:44 2014
@@ -497,14 +497,22 @@
 };
 
 const struct lconv_offset lconv_integers[] = {
-    {"int_frac_digits",   offsetof(struct lconv, int_frac_digits)},
-    {"frac_digits",       offsetof(struct lconv, frac_digits)},
-    {"p_cs_precedes",     offsetof(struct lconv, p_cs_precedes)},
-    {"p_sep_by_space",    offsetof(struct lconv, p_sep_by_space)},
-    {"n_cs_precedes",     offsetof(struct lconv, n_cs_precedes)},
-    {"n_sep_by_space",    offsetof(struct lconv, n_sep_by_space)},
-    {"p_sign_posn",       offsetof(struct lconv, p_sign_posn)},
-    {"n_sign_posn",       offsetof(struct lconv, n_sign_posn)},
+    {"int_frac_digits",     offsetof(struct lconv, int_frac_digits)},
+    {"frac_digits",         offsetof(struct lconv, frac_digits)},
+    {"p_cs_precedes",       offsetof(struct lconv, p_cs_precedes)},
+    {"p_sep_by_space",      offsetof(struct lconv, p_sep_by_space)},
+    {"n_cs_precedes",       offsetof(struct lconv, n_cs_precedes)},
+    {"n_sep_by_space",      offsetof(struct lconv, n_sep_by_space)},
+    {"p_sign_posn",         offsetof(struct lconv, p_sign_posn)},
+    {"n_sign_posn",         offsetof(struct lconv, n_sign_posn)},
+#ifdef HAS_LC_MONETARY_2008
+    {"int_p_cs_precedes",   offsetof(struct lconv, int_p_cs_precedes)},
+    {"int_p_sep_by_space",  offsetof(struct lconv, int_p_sep_by_space)},
+    {"int_n_cs_precedes",   offsetof(struct lconv, int_n_cs_precedes)},
+    {"int_n_sep_by_space",  offsetof(struct lconv, int_n_sep_by_space)},
+    {"int_p_sign_posn",     offsetof(struct lconv, int_p_sign_posn)},
+    {"int_n_sign_posn",     offsetof(struct lconv, int_n_sign_posn)},
+#endif
     {NULL, 0}
 };
 
--- ./ext/POSIX/t/posix.t.orig	Mon Jan  6 14:46:45 2014
+++ ./ext/POSIX/t/posix.t	Wed Jul  9 16:31:44 2014
@@ -8,7 +8,7 @@
     }
 }
 
-use Test::More tests => 109;
+use Test::More tests => 115;
 
 use POSIX qw(fcntl_h signal_h limits_h _exit getcwd open read strftime write
 	     errno localeconv dup dup2 lseek access);
@@ -330,8 +330,24 @@
 	}
     }
 
-    foreach (qw(int_frac_digits frac_digits p_cs_precedes p_sep_by_space
-		n_cs_precedes n_sep_by_space p_sign_posn n_sign_posn)) {
+    my @lconv = qw(
+        int_frac_digits frac_digits
+        p_cs_precedes   p_sep_by_space
+        n_cs_precedes   n_sep_by_space
+        p_sign_posn     n_sign_posn
+    );
+
+    SKIP: {
+        skip('No HAS_LC_MONETARY_2008', 6) unless $Config{d_lc_monetary_2008};
+
+        push @lconv, qw(
+            int_p_cs_precedes int_p_sep_by_space
+            int_n_cs_precedes int_n_sep_by_space
+            int_p_sign_posn   int_n_sign_posn
+        );
+    }
+        
+    foreach (@lconv) {
     SKIP: {
 	    skip("localeconv has no result for $_", 1)
 		unless exists $conv->{$_};
--- ./NetWare/config.wc.orig	Mon Jan  6 14:46:42 2014
+++ ./NetWare/config.wc	Wed Jul  9 16:31:44 2014
@@ -285,6 +285,7 @@
 d_isnan='undef'
 d_isnanl='undef'
 d_killpg='undef'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='define'
 d_libm_lib_version='undef'
--- ./plan9/config_sh.sample.orig	Mon Jan  6 14:46:45 2014
+++ ./plan9/config_sh.sample	Wed Jul  9 16:31:44 2014
@@ -296,6 +296,7 @@
 d_isnan='undef'
 d_isnanl='undef'
 d_killpg='undef'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='define'
 d_libm_lib_version='0'
--- ./Porting/config.sh.orig	Mon Jan  6 14:46:42 2014
+++ ./Porting/config.sh	Wed Jul  9 16:31:44 2014
@@ -310,6 +310,7 @@
 d_isnan='define'
 d_isnanl='define'
 d_killpg='define'
+d_lc_monetary_2008='undef'
 d_lchown='define'
 d_ldbl_dig='define'
 d_libm_lib_version='define'
--- ./symbian/config.sh.orig	Mon Jan  6 14:46:45 2014
+++ ./symbian/config.sh	Wed Jul  9 16:31:44 2014
@@ -242,6 +242,7 @@
 d_isnan='undef'
 d_isnanl='undef'
 d_killpg='undef'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='undef'
 d_libm_lib_version='undef'
--- ./uconfig.h.orig	Mon Jan  6 14:46:46 2014
+++ ./uconfig.h	Wed Jul  9 16:31:44 2014
@@ -226,6 +226,12 @@
  */
 /*#define HAS_LOCALECONV	/ **/
 
+/* HAS_LC_MONETARY_2008:
+ *	This symbol, if defined, indicates that the localeconv routine is
+ *	available and has the additional members added in POSIX 1003.1-2008.
+ */
+/*#define HAS_LC_MONETARY_2008	/ **/
+
 /* HAS_LOCKF:
  *	This symbol, if defined, indicates that the lockf routine is
  *	available to do file locking.
@@ -4753,6 +4759,6 @@
 #endif
 
 /* Generated from:
- * 2c9dc3f21d37b1665f6a59dfc6d79e6cb08bdf36a9c3e427d11d6b9ddffe2439 config_h.SH
- * 26ab9b4aa382d32761cb91084ba59e7e4b190799502a43366ccb3d2f584783ca uconfig.sh
+ * 94fbe0b36e9c9bb105c5b4a7a59bd71958b11736c11b971b785d313682d3b672 config_h.SH
+ * 3d49fdf749dbdf4cf5bbef77b3bec93cddf5b7e66553df67d215d570855d6545 uconfig.sh
  * ex: set ro: */
--- ./uconfig.sh.orig	Mon Jan  6 14:46:46 2014
+++ ./uconfig.sh	Wed Jul  9 16:31:44 2014
@@ -235,6 +235,7 @@
 d_isnan='undef'
 d_isnanl='undef'
 d_killpg='undef'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='undef'
 d_libm_lib_version='undef'
--- ./uconfig64.sh.orig	Mon Jan  6 14:46:46 2014
+++ ./uconfig64.sh	Wed Jul  9 16:31:44 2014
@@ -236,6 +236,7 @@
 d_isnan='undef'
 d_isnanl='undef'
 d_killpg='undef'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='undef'
 d_libm_lib_version='undef'
--- ./win32/config.ce.orig	Mon Jan  6 14:46:46 2014
+++ ./win32/config.ce	Wed Jul  9 16:31:44 2014
@@ -283,6 +283,7 @@
 d_isnan='define'
 d_isnanl='undef'
 d_killpg='undef'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='define'
 d_libm_lib_version='undef'
--- ./win32/config.gc.orig	Mon Jan  6 14:46:46 2014
+++ ./win32/config.gc	Wed Jul  9 16:31:44 2014
@@ -283,6 +283,7 @@
 d_isnan='define'
 d_isnanl='undef'
 d_killpg='define'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='define'
 d_libm_lib_version='undef'
--- ./win32/config.vc.orig	Mon Jan  6 14:46:46 2014
+++ ./win32/config.vc	Wed Jul  9 16:31:44 2014
@@ -283,6 +283,7 @@
 d_isnan='define'
 d_isnanl='undef'
 d_killpg='define'
+d_lc_monetary_2008='undef'
 d_lchown='undef'
 d_ldbl_dig='define'
 d_libm_lib_version='undef'
