--- ./MANIFEST.orig	Tue Jun 30 01:47:24 2026
+++ ./MANIFEST	Thu Aug 20 01:40:00 2026
@@ -4770,7 +4770,6 @@ ext/B/B/Showlex.pm			Compiler Showlex backend
 ext/B/B/Terse.pm			Compiler Terse backend
 ext/B/B/Xref.pm				Compiler Xref backend
 ext/B/hints/darwin.pl			Hints for named architecture
-ext/B/hints/openbsd.pl			Hints for named architecture
 ext/B/Makefile.PL			Compiler backend makefile writer
 ext/B/O.pm				Compiler front-end module (-MO=...)
 ext/B/t/b.t				See if B works
