--- ./MANIFEST.orig	Tue Mar 18 10:48:51 2014
+++ ./MANIFEST	Tue Mar 25 00:54:11 2014
@@ -3577,7 +3577,6 @@
 ext/B/B/Xref.pm		Compiler Xref backend
 ext/B/B.xs		Compiler backend external subroutines
 ext/B/hints/darwin.pl	Hints for named architecture
-ext/B/hints/openbsd.pl	Hints for named architecture
 ext/B/Makefile.PL	Compiler backend makefile writer
 ext/B/O.pm		Compiler front-end module (-MO=...)
 ext/B/t/b.t		See if B works
