--- ./MANIFEST.orig	Mon Mar  4 08:16:21 2013
+++ ./MANIFEST	Wed Mar 13 11:57:33 2013
@@ -3626,7 +3626,6 @@
 ext/B/B/Xref.pm		Compiler Xref backend
 ext/B/B.xs		Compiler backend external subroutines
 ext/B/hints/darwin.pl	Hints for named architecture
-ext/B/hints/openbsd.pl	Hints for named architecture
 ext/B/Makefile.PL	Compiler backend makefile writer
 ext/B/O.pm		Compiler front-end module (-MO=...)
 ext/B/t/b.t		See if B works
