--- ./ext/DynaLoader/dl_dlopen.xs.orig	Sun Sep 14 04:31:01 2014
+++ ./ext/DynaLoader/dl_dlopen.xs	Tue Nov 25 19:21:00 2014
@@ -170,8 +170,8 @@ dl_load_file(filename, flags=0)
     char pathbuf[PATH_MAX + 2];
     if (*filename != '/' && strchr(filename, '/')) {
 	if (getcwd(pathbuf, PATH_MAX - strlen(filename))) {
-	    strcat(pathbuf, "/");
-	    strcat(pathbuf, filename);
+	    my_strlcat(pathbuf, "/", sizeof(pathbuf));
+	    my_strlcat(pathbuf, filename, sizeof(pathbuf));
 	    filename = pathbuf;
 	}
     }
