--- ./cpan/Text-Tabs/CHANGELOG.orig	Mon Jan  6 14:46:44 2014
+++ ./cpan/Text-Tabs/CHANGELOG	Sat Mar 22 14:04:24 2014
@@ -1,4 +1,20 @@
 
+= 2013/05/23
+
+Change module 'NAME'
+
+= 2013/05/22
+
+Typos
+
+= 2013/04/26
+
+Minor test suite fixes - bug 81698.
+
+Fixed bug 79766 -- an extraneous "=" in a regex.
+
+Changed the license to qualify as an "open source" license.
+
 = 2012/08/18
 
 Packaging fix.
--- ./cpan/Text-Tabs/lib/Text/Tabs.pm.orig	Mon Jan  6 14:46:44 2014
+++ ./cpan/Text-Tabs/lib/Text/Tabs.pm	Sat Mar 22 14:04:24 2014
@@ -1,4 +1,3 @@
-
 package Text::Tabs;
 
 require Exporter;
@@ -7,7 +6,7 @@
 @EXPORT = qw(expand unexpand $tabstop);
 
 use vars qw($VERSION $SUBVERSION $tabstop $debug);
-$VERSION = 2012.0818;
+$VERSION = 2013.0523;
 $SUBVERSION = 'modern';
 
 use strict;
@@ -157,7 +156,7 @@
     print unexpand $_;
   }
 
-Instead of the shell's C<expand> comand, use:
+Instead of the shell's C<expand> command, use:
 
   perl -MText::Tabs -n -e 'print expand $_'
 
@@ -186,7 +185,8 @@
 
 Copyright (C) 1996-2002,2005,2006 David Muir Sharnoff.  
 Copyright (C) 2005 Aristotle Pagaltzis 
-Copyright (C) 2012 Google, Inc.
+Copyright (C) 2012-2013 Google, Inc.
 This module may be modified, used, copied, and redistributed at your own risk.
-Publicly redistributed modified versions must use a different name.
-
+Although allowed by the preceding license, please do not publicly
+redistribute modified versions of this code with the name "Text::Tabs"
+unless it passes the unmodified Text::Tabs test suite.
--- ./cpan/Text-Tabs/lib/Text/Wrap.pm.orig	Mon Jan  6 14:46:44 2014
+++ ./cpan/Text-Tabs/lib/Text/Wrap.pm	Sat Mar 22 14:04:24 2014
@@ -7,7 +7,7 @@
 @EXPORT = qw(wrap fill);
 @EXPORT_OK = qw($columns $break $huge);
 
-$VERSION = 2012.0818;
+$VERSION = 2013.0523;
 $SUBVERSION = 'modern';
 
 use 5.010_000;
@@ -64,7 +64,7 @@
 				? unexpand($nl . $lead . $1)
 				: $nl . $lead . $1;
 			$remainder = $2;
-		} elsif ($huge eq 'wrap' && $t =~ /\G((?:(?!=[^\n])\X){$ll})/gc) {
+		} elsif ($huge eq 'wrap' && $t =~ /\G((?:(?=[^\n])\X){$ll})/gc) {
 			$r .= $unexpand 
 				? unexpand($nl . $lead . $1)
 				: $nl . $lead . $1;
@@ -293,7 +293,8 @@
 =head1 LICENSE
 
 Copyright (C) 1996-2009 David Muir Sharnoff.  
-Copyright (C) 2012 Google, Inc.
+Copyright (C) 2012-2013 Google, Inc.
 This module may be modified, used, copied, and redistributed at your own risk.
-Publicly redistributed modified versions must use a different name.
-
+Although allowed by the preceding license, please do not publicly
+redistribute modified versions of this code with the name "Text::Wrap"
+unless it passes the unmodified Text::Wrap test suite.
--- ./cpan/Text-Tabs/t/Tabs-ElCid.t.orig	Mon Jan  6 14:46:44 2014
+++ ./cpan/Text-Tabs/t/Tabs-ElCid.t	Sat Mar 22 14:04:24 2014
@@ -2,7 +2,7 @@
 
 BEGIN {
 	if ($] <= 5.010) {
-		print "1..0 # skip this test requires perl 5.010 or greater\n";
+		print "1..0 # skip this test requires perl 5.010 or greater ($])\n";
 		exit 0;
 	}
 }
@@ -105,8 +105,8 @@
 
 sub check_data { 
 
-    local $_;
     binmode(DATA, ":utf8") || die "can't binmode DATA to utf8: $!";
+    local($_);
     while ( <DATA> ) {
 
 	my $bad = 0;
--- ./cpan/Text-Tabs/t/Wrap-JLB.t.orig	Mon Jan  6 14:46:44 2014
+++ ./cpan/Text-Tabs/t/Wrap-JLB.t	Sat Mar 22 14:04:24 2014
@@ -87,7 +87,7 @@
 sub check_data { 
 
     binmode(DATA, ":utf8") || die "can't binmode DATA to utf8: $!";
-    local $_;
+    local($_);
     while ( <DATA> ) {
 
 	my $bad = 0;
--- ./cpan/Text-Tabs/t/79766.t.orig	Sat Mar 22 14:04:24 2014
+++ ./cpan/Text-Tabs/t/79766.t	Sat Mar 22 14:04:24 2014
@@ -0,0 +1,18 @@
+#!/usr/bin/perl -I.
+
+use warnings;
+use strict;
+use Text::Wrap;
+use Test::More tests => 2;
+
+my $r;
+my $s = q{xx xxxxxxxx xxxxxxxxx xx xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=xxxxxxxxxxxxxxxxxxxxxxxx};
+
+eval { $r = wrap("", "", $s) };
+ok(! $@, $@);
+is($r, "xx xxxxxxxx xxxxxxxxx xx
+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=xxxxxxxxxxxxx
+xxxxxxxxxxx", "match");
+
--- ./MANIFEST.orig	Mon Jan  6 14:46:42 2014
+++ ./MANIFEST	Sat Mar 22 14:04:24 2014
@@ -2672,6 +2672,7 @@
 cpan/Text-Tabs/lib/Text/Wrap.pm		Paragraph formatter
 cpan/Text-Tabs/t/37000.t		See if Text::Tabs is working
 cpan/Text-Tabs/t/39548.t		See if Text::Tabs is working
+cpan/Text-Tabs/t/79766.t		See if Text::Tabs is working
 cpan/Text-Tabs/t/belg4mit.t		See if Text::Tabs is working
 cpan/Text-Tabs/t/dandv.t		See if Text::Tabs is working
 cpan/Text-Tabs/t/fill.t			See if Text::Wrap::fill works
--- ./Porting/Maintainers.pl.orig	Mon Jan  6 14:46:42 2014
+++ ./Porting/Maintainers.pl	Sat Mar 22 14:04:24 2014
@@ -1871,7 +1871,7 @@
 
     'Text-Tabs+Wrap' => {
         'MAINTAINER'   => 'muir',
-        'DISTRIBUTION' => 'MUIR/modules/Text-Tabs+Wrap-2012.0818.tar.gz',
+        'DISTRIBUTION' => 'MUIR/modules/Text-Tabs+Wrap-2013.0523.tar.gz',
         'FILES'        => q[cpan/Text-Tabs],
         'EXCLUDED'   => [
             qr/^lib\.old/,
