--- ./hints/openbsd.sh.orig	Mon Sep 20 19:38:30 2021
+++ ./hints/openbsd.sh	Thu Dec  8 04:41:23 2022
@@ -101,6 +101,13 @@ m88k-3.4)
    ;;
 esac
 
+# Special per-arch specific ccflags
+case "${ARCH}-${osvers}" in
+    vax-*)
+    ccflags="-DUSE_PERL_ATOF=0 $ccflags"
+    ;;
+esac
+
 # This script UU/usethreads.cbu will get 'called-back' by Configure 
 # after it has prompted the user for whether to use threads.
 cat > UU/usethreads.cbu <<'EOCBU'
