--- ./hints/openbsd.sh.orig	Sun Oct 20 06:31:29 2013
+++ ./hints/openbsd.sh	Wed Mar 26 23:25:06 2014
@@ -15,11 +15,6 @@
     ;;
 esac
 
-# malloc wrap works
-case "$usemallocwrap" in
-'') usemallocwrap='define' ;;
-esac
-
 # Currently, vfork(2) is not a real win over fork(2).
 usevfork="$undef"
 
@@ -83,6 +78,14 @@
 	i_dbm=$undef
 	;;
 esac
+
+# malloc wrap causes problems on m68k
+if [ X"$usemallocwrap" = X"" ]; then
+	case "${ARCH}" in
+	m68k) usemallocwrap="$undef" ;;
+	*)    usemallocwrap="define" ;;
+	esac
+fi
 
 # OpenBSD doesn't need libcrypt but many folks keep a stub lib
 # around for old NetBSD binaries.
