--- ./hints/openbsd.sh.orig	Wed May 13 13:19:29 2015
+++ ./hints/openbsd.sh	Thu Sep 17 23:02:19 2015
@@ -101,6 +101,13 @@ m88k-3.4)
    ;;
 esac
 
+# Special per-arch specific ccflags
+case "${ARCH}-${osvers}" in
+    vax-*)
+    ccflags="-DUSE_PERL_ATOF=0 $ccflags"
+    ;;
+esac
+
 # This script UU/usethreads.cbu will get 'called-back' by Configure 
 # after it has prompted the user for whether to use threads.
 cat > UU/usethreads.cbu <<'EOCBU'
