--- ./hints/openbsd.sh.orig	Fri Jun 27 07:48:18 2025
+++ ./hints/openbsd.sh	Wed Jul  2 16:20:00 2025
@@ -101,6 +101,13 @@ m88k-3.4)
    ;;
 esac
 
+# Special per-arch specific ccflags
+case "${ARCH}-${osvers}" in
+    vax-*)
+    ccflags="-DUSE_PERL_ATOF=0 $ccflags"
+    ;;
+esac
+
 # This script UU/usethreads.cbu will get 'called-back' by Configure 
 # after it has prompted the user for whether to use threads.
 cat > UU/usethreads.cbu <<'EOCBU'
