diff -Nrux *.orig ../perl-5.16.0.orig/Makefile.SH ./Makefile.SH
--- ../perl-5.16.0.orig/Makefile.SH	Mon May 14 12:49:21 2012
+++ ./Makefile.SH	Sun Jun  3 22:52:15 2012
@@ -404,7 +404,7 @@
 CCCMDSRC = `sh $(shellflags) cflags "optimize='$(OPTIMIZE)'" $<`
 
 CONFIGPM_FROM_CONFIG_SH = lib/Config.pm lib/Config_heavy.pl
-CONFIGPM = $(CONFIGPM_FROM_CONFIG_SH) lib/Config_git.pl
+CONFIGPM = $(CONFIGPM_FROM_CONFIG_SH)
 
 CONFIGPOD = lib/Config.pod
 
diff -ru ../perl-5.16.0.orig/Makefile.SH ./Makefile.SH
--- ../perl-5.16.0.orig/Makefile.SH	Mon May 14 12:49:21 2012
+++ ./Makefile.SH	Mon Jun  4 20:10:11 2012
@@ -436,7 +436,7 @@
 ## In the following dollars and backticks do not need the extra backslash.
 $spitshell >>$Makefile <<'!NO!SUBS!'
 
-private = preplibrary $(CONFIGPM) $(CONFIGPOD) lib/ExtUtils/Miniperl.pm git_version.h lib/buildcustomize.pl
+private = preplibrary $(CONFIGPM) $(CONFIGPOD) lib/ExtUtils/Miniperl.pm lib/buildcustomize.pl
 
 # Files to be built with variable substitution before miniperl
 # is available.
diff -Nrux *.orig ../perl-5.16.0.orig/Makefile.SH ./Makefile.SH
--- ../perl-5.16.0.orig/Makefile.SH	Mon May 14 12:49:21 2012
+++ ./Makefile.SH	Sun Jun  3 23:18:26 2012
@@ -561,12 +561,13 @@
 
 .PHONY: all translators utilities
 
+# OpenBSD uses pre-generated lib/Config_git.pl and git_version.h files
 # Both git_version.h and lib/Config_git.pl are built
 # by make_patchnum.pl.
-git_version.h: lib/Config_git.pl
-
-lib/Config_git.pl: $(MINIPERL_EXE) make_patchnum.pl
-	$(MINIPERL) make_patchnum.pl
+#git_version.h: lib/Config_git.pl
+#
+#lib/Config_git.pl: $(MINIPERL_EXE) make_patchnum.pl
+#	$(MINIPERL) make_patchnum.pl
 
 # make sure that we recompile perl.c if the git version changes
 perl$(OBJ_EXT): git_version.h
diff -Nrux *.orig ../perl-5.16.0.orig/Makefile.SH ./Makefile.SH
--- ../perl-5.16.0.orig/Makefile.SH	Mon May 14 12:49:21 2012
+++ ./Makefile.SH	Mon Jun  4 00:14:48 2012
@@ -1263,14 +1263,13 @@
 	-rm -f perl.Addrs perl.Counts t/perl.Addrs t/perl.Counts *perl.xok
 	-rm -f cygwin.c libperl*.def libperl*.dll cygperl*.dll *.exe.stackdump
 	-rm -f $(PERL_EXE) $(MINIPERL_EXE) $(LIBPERL) libperl.* microperl
-	-rm -f config.arch config.over $(DTRACE_H) runtests
+	-rm -f $(DTRACE_H) runtests
 
 # Do not 'make _tidy' directly.
 _tidy:
 	-cd pod; $(LDLIBPTH) $(MAKE) clean
 	-cd utils; $(LDLIBPTH) $(MAKE) clean
 	-cd x2p; $(LDLIBPTH) $(MAKE) clean
-	-rm -f lib/Config_git.pl git_version.h
 	-@for x in $(DYNALOADER) $(dynamic_ext) $(static_ext) $(nonxs_ext) ; do \
 	$(MINIPERL) make_ext.pl --target=clean $$x MAKE=$(MAKE) ; \
 	done
